<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Ben Mescher" />


<title>Analysis of CML Microarray Data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Ben Mescher SBMI5330 Project Site</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="exploring_clusterprofiler.html">Exploring ClusterProfiler</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Analysis of CML Microarray Data</h1>
<h4 class="author">Ben Mescher</h4>

</div>


<p>Final, formal, paper write up is saved here: <a href="https://docs.google.com/document/d/1sDRhekkbcrHK-mQRN2IN-v70ZD0oT5KMHAc2F7OPLVA/edit?usp=sharing" class="uri">https://docs.google.com/document/d/1sDRhekkbcrHK-mQRN2IN-v70ZD0oT5KMHAc2F7OPLVA/edit?usp=sharing</a></p>
<p>Preliminary goals: Familiarize with the GEOquery and BioConductor data structure for microarray expression data. Explore significantly differing genes using limma contrasts, then plot expression profiles for the top genes. Review original paper for more info about the data samples (some samples weren’t used, some should be excluded due to not being clearly in an advanced state [remission cases, etc.]) For the top genes, plot common pathways or biologic function (using ClusterProfiler?. <a href="http://bioconductor.org/packages/release/bioc/vignettes/clusterProfiler/inst/doc/clusterProfiler.html#go-classification" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/clusterProfiler/inst/doc/clusterProfiler.html#go-classification</a> )</p>
<p>Next: see if can recreate Radich’ conclusions, which compared chronic with accelerated + blast crisis (subtract healthy CD34+ from all disease samples); identifying the ‘phase reporter’ set of genes.</p>
<div id="analyzing-microarray-data-for-radich-2006" class="section level2">
<h2>Analyzing Microarray Data for Radich (2006)</h2>
<div id="step-1-download-and-format-geo-data-with-geo2r" class="section level3">
<h3>Step 1: Download and format GEO Data with GEO2R</h3>
<ul>
<li>Start with some scripts as provided by GEO2R for the Radich (2006) microarray repositories. Will pull data for the Deininger (2009) repository afterwards.</li>
</ul>
<div id="geoquery-will-return-an-expressionset-object-which-holds-the-following" class="section level4">
<h4>GEOquery will return an “ExpressionSet” object, which holds the following:</h4>
<ul>
<li>experimentData - metadata on the whole experiment</li>
<li>assayData - large matrix of numeric expression values, accessed with expres()</li>
<li>phenoData - dataframe on each run (sample), accessed with pData()</li>
<li>featureData - dataframe on each of the 20k+ genes (ID can join to assay data, also has GO IDs, etc.)</li>
</ul>
<pre class="r"><code>#Download GEO data for GSE4170 into a table &quot;gset&quot;. 19MB
suppressMessages(gset &lt;- getGEO(&quot;GSE4170&quot;, GSEMatrix =TRUE, AnnotGPL=TRUE)) 
#Pull column names from the platform for this experiment: Rosetta/Merck Human 25k v2.2.1 microarray (GPL2029)
#getGEO returns a list of experiments. we are only interested the 1st (and only) one, so set gset = gset[[1]]
if (length(gset) &gt; 1) idx &lt;- grep(&quot;GPL2029&quot;, attr(gset, &quot;names&quot;)) else idx &lt;- 1
gset &lt;- gset[[idx]]

gset</code></pre>
<pre><code>## ExpressionSet (storageMode: lockedEnvironment)
## assayData: 24481 features, 119 samples 
##   element names: exprs 
## protocolData: none
## phenoData
##   sampleNames: GSM95115 GSM95116 ... GSM95234 (119 total)
##   varLabels: title geo_accession ... data_row_count (34 total)
##   varMetadata: labelDescription
## featureData
##   featureNames: 10012665102 10012665103 ... 10017963787 (24481
##     total)
##   fvarLabels: ID Gene title ... GO:Component ID (21 total)
##   fvarMetadata: Column Description labelDescription
## experimentData: use &#39;experimentData(object)&#39;
##   pubMedIds: 16477019 
## Annotation: GPL2029</code></pre>
<p>The microarray expression values range from -2 to +2. With the median at 00.54, I believe these are all log10 of the microarray.</p>
<pre class="r"><code>quantile(exprs(gset), c(0., 0.25, 0.5, 0.75, 0.99, 1.0), na.rm=T)</code></pre>
<pre><code>##      0%     25%     50%     75%     99%    100% 
## -2.0000 -0.1149  0.0056  0.1352  1.7340  2.0000</code></pre>
<p>Perform some pre-processing of the data, formating column names and such. This code is largely taken from GEO2R, though I went through to make sure it’s applicable here. Repurpose the “description” slot (attribute) to categorize the samples into user-defined groups (I chose these myself in the GEO2R web tool).</p>
<pre class="r"><code>#(Code block will be hidden in final HTML of this notebook).</code></pre>
<pre class="r"><code>#phenotypes data (metadata) on the first 10 samples
pData(gset)[1:10,c(&quot;title&quot;,&quot;geo_accession&quot;,&quot;description&quot;,&quot;description.1&quot;,  &quot;data_row_count&quot;)]</code></pre>
<pre><code>##                title geo_accession description           description.1
## GSM95115    CD34-001      GSM95115          G0              type: cd34
## GSM95116    CD34-003      GSM95116          G0              type: cd34
## GSM95117    CD34-004      GSM95117          G0              type: cd34
## GSM95118    CD34-005      GSM95118          G0              type: cd34
## GSM95119    CD34-006      GSM95119          G0              type: cd34
## GSM95120 CD34-920082      GSM95120          G0              type: cd34
## GSM95121      AP-001      GSM95121          G2           type: ap-cyto
## GSM95122      AP-002      GSM95122          G2           type: ap-cyto
## GSM95123      AP-005      GSM95123          G2 type: accelerated phase
## GSM95124      AP-007      GSM95124          G2           type: ap-cyto
##          data_row_count
## GSM95115          24481
## GSM95116          24481
## GSM95117          24481
## GSM95118          24481
## GSM95119          24481
## GSM95120          24481
## GSM95121          24481
## GSM95122          24481
## GSM95123          24481
## GSM95124          24481</code></pre>
<pre class="r"><code>#example numeric expression data for 7 genes in each of the first 5 samples
exprs(gset)[1:7,1:5]</code></pre>
<pre><code>##             GSM95115 GSM95116 GSM95117 GSM95118 GSM95119
## 10012665102   0.1708   0.2028  -0.0109   1.2057   0.2943
## 10012665103  -0.1077  -0.0304   0.0087   0.5269   0.4897
## 10012665104   0.0178  -0.0665  -0.0095   0.1281  -0.2027
## 10012665105   0.0764   0.0033   0.1035   1.1692   1.6524
## 10012665106  -0.1175  -0.0590  -0.0316   1.0877   1.2177
## 10012665107  -0.2158  -0.1017  -0.0862  -0.2702  -0.3925
## 10012665108   0.1444  -0.0047   0.1787   0.1405   0.0230</code></pre>
</div>
</div>
<div id="step-2-fit-to-a-linear-model-pairwise-differential-expression" class="section level3">
<h3>Step 2: Fit to a linear model, Pairwise Differential Expression</h3>
<p>Use limma (lmFit, from Bioconductor) to fit linear model for each gene (fit expression data, use the model matrix.) Then build on the initial model into a contrasts model, with empirical Bayes test statistics:</p>
<p>Input the lmFit and the makeContrasts 4x4 matrix into “contrasts.fit”, which will estimate coefficients and standard errors for the contrasts. Use eBayes to calculate t-statistics, F-statistics by empirical Bayes moderation. This “fit2” model will save statistics for each gene, for each of the 4 comparisons (so it’s a 20k x 4 matrix).</p>
<p>I think “topTable” only retrieves the first column (which would be G5 to G0 comparison here.)</p>
<pre class="r"><code>#build a linear model... 
fit &lt;- lmFit(gset, design)</code></pre>
<pre><code>## Warning: Partial NA coefficients for 607 probe(s)</code></pre>
<pre class="r"><code>cont.matrix &lt;- makeContrasts(G1-G0, G2-G0, G3-G0, G3-G2, G3-G1, levels=design) #for 4 groups: 0,1,2,5. 
#^I think it is here you can group Blast Crisis and Accelerated groups into one and compare vs. control group!
fit2 &lt;- contrasts.fit(fit, cont.matrix)
fit2 &lt;- eBayes(fit2, 0.01)
tT &lt;- topTable(fit2, adjust=&quot;fdr&quot;, sort.by=&quot;B&quot;, number=250) #sort by the lods or B-statistic</code></pre>
<div id="top-20-most-differentially-expressed-genes-in-pairwise-group-comparisons" class="section level4">
<h4>Top 20 Most Differentially Expressed Genes in Pairwise Group Comparisons</h4>
<p>View data for the 20 most statistically relevant genes that were differentially expressed in the following groups: * G0 is control CD34 * G1 is Chronic Phase * G2 is Accelerated Phase * G3 is Blast Crisis</p>
<p>This first round, I only did 5 pairwise contrasts for the following: G1 and G0 (chronic and control) G2 and G0 (accelerated and control) G3 and G0 (crisis and control) G3 and G2 (crisis vs accelerated. not expected to be largely different) G3 and G1 (chronic and blast crisis, least and most advanced disease states)</p>
<pre class="r"><code>#by default, top table saves a dataframe with all columns of the gene feature data in the expression dataset. 
#subset top table here down to only the columns we care about:
subset(tT, select=c(&quot;P.Value&quot;,&quot;Gene.symbol&quot;
                    ,&quot;G1...G0&quot;,&quot;G2...G0&quot;,&quot;G3...G0&quot;,&quot;G3...G2&quot;,&quot;G3...G1&quot;
                    ,&quot;AveExpr&quot;))[1:20,]</code></pre>
<pre><code>##                  P.Value Gene.symbol    G1...G0      G2...G0     G3...G0
## 10012672278 4.326511e-32         GAK  0.4329386  0.233988095  0.08015417
## 10012667916 6.967740e-29       VDAC2 -0.5550561 -0.257616667 -0.03490917
## 10012685105 1.477568e-28    FLJ40142 -0.7656912 -0.257484615 -0.07686500
## 10012688029 2.934467e-28       ITGAM  1.6470175  1.517516667  0.61402417
## 10012684680 4.354013e-28       ITGB2  1.1088035  0.903050000  0.39609231
## 10012674495 9.584159e-28       BCAT1 -0.9783939 -0.685314286 -0.11528250
## 10012685898 1.084776e-27        TCP1 -0.6470132 -0.473578571 -0.24238750
## 10012676178 1.636838e-27       CD109 -1.2332658 -0.468750000 -0.14862000
## 10012672593 3.100368e-27       SOCS2 -0.5935421 -0.009833333  0.42497667
## 10012672247 3.773066e-27         FGR  1.2933648  1.091309524  0.44514314
## 10012673694 3.886184e-27       TFB2M -0.7091298 -0.391792857 -0.17498000
## 10012681110 5.329463e-27       BCAT1 -0.9297307 -0.647135714 -0.08711250
## 10012670381 5.891002e-27        AVEN -0.5644053 -0.369216667 -0.08593917
## 10012672906 6.976105e-27      TCIRG1  0.8570658  0.614367949  0.30966083
## 10012688416 7.941777e-27         HK3  1.6237976  1.383229487  0.48534583
## 10012685376 1.371646e-26       ABCA7  0.7088792  0.522623810  0.14045667
## 10012673073 1.795052e-26      TCIRG1  0.8500711  0.612337179  0.30499833
## 10012672263 3.606272e-26       RIOK1 -0.5926465 -0.296228571 -0.18592500
## 10012682071 4.593546e-26     SLC15A3  1.3998886  1.321328571  0.71454750
## 10012673381 9.989036e-26       TTC31  0.1806114  0.022307143  0.01428250
##                  G3...G2    G3...G1     AveExpr
## 10012672278 -0.153833929 -0.3527844 -0.17654615
## 10012667916  0.222707500  0.5201470  0.27289402
## 10012685105  0.180619615  0.6888262  0.49049397
## 10012688029 -0.903492500 -1.0329934 -0.60546752
## 10012684680 -0.506957692 -0.7127112 -0.41529914
## 10012674495  0.570031786  0.8631114  0.38858034
## 10012685898  0.231191071  0.4046257  0.23700342
## 10012676178  0.320130000  1.0846458  0.58332821
## 10012672593  0.434810000  1.0185188  0.54008632
## 10012672247 -0.646166387 -0.8482216 -0.45909252
## 10012673694  0.216812857  0.5341498  0.30592308
## 10012681110  0.560023214  0.8426182  0.43178718
## 10012670381  0.283277500  0.4784661  0.32053932
## 10012672906 -0.304707115 -0.5474050 -0.29940776
## 10012688416 -0.897883654 -1.1384518 -0.63228348
## 10012685376 -0.382167143 -0.5684225 -0.29499052
## 10012673073 -0.307338846 -0.5450727 -0.29528017
## 10012672263  0.110303571  0.4067215  0.24791453
## 10012682071 -0.606781071 -0.6853411 -0.34455556
## 10012673381 -0.008024643 -0.1663289 -0.08940769</code></pre>
</div>
</div>
<div id="next-look-at-actual-expression-values-for-the-top-3" class="section level3">
<h3>Next, look at actual expression values for the top 3:</h3>
<p>Look up info on the top gene, GAK:</p>
<pre class="r"><code>#transpose the feature (gene) data for the GAK gene...
t(gset@featureData@data[with(gset@featureData@data,Gene.symbol==&quot;GAK&quot;)
                        ,c(&quot;ID&quot;,&quot;Gene.symbol&quot;,&quot;Gene.title&quot;,&quot;Chromosome.location&quot;,&quot;GO.Process&quot;)])</code></pre>
<pre><code>##                     10012672278                                      
## ID                  &quot;10012672278&quot;                                    
## Gene.symbol         &quot;GAK&quot;                                            
## Gene.title          &quot;cyclin G associated kinase&quot;                     
## Chromosome.location &quot;4p16&quot;                                           
## GO.Process          &quot;cell cycle///protein amino acid phosphorylation&quot;</code></pre>
<p>Look at expression values for some of the top genes. Accelerated and blast crisis groups are more similar to the healthy CD34 group than the chronic phase samples. Radich 2006 posits this is because advanced CML cells have moved past dependency on just these specific pathways.</p>
<pre class="r"><code>x = 1
#instead of having to look up IDs, just take the xth value of the TopTable...
plot(exprs(gset)[toString(tT[x,1]),] ~ fl #plot expressions for xth most differentially expressed gene
     ,main = paste(tT[x,&quot;Gene.symbol&quot;] , &quot; (#&quot;, x, &quot; most significantly differentially expressed gene)&quot;)
     ,xlab=&quot;Groups: G0 (Control), G1 (Chronic), G2 (Accelerated), G3 (Crises)&quot;,ylab=&quot;&quot;
     ,sub = paste(&quot;GO Process:&quot;,tT[x,&quot;GO.Process&quot;]))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<pre class="r"><code>x = 2
#instead of having to look up IDs, just take the xth value of the TopTable...
plot(exprs(gset)[toString(tT[x,1]),] ~ fl #plot expressions for xth most differentially expressed gene
     ,main = paste(tT[x,&quot;Gene.symbol&quot;] , &quot; (#&quot;, x, &quot; most significantly differentially expressed gene)&quot;)
     ,xlab=&quot;Groups: G0 (Control), G1 (Chronic), G2 (Accelerated), G3 (Crises)&quot;,ylab=&quot;&quot;
     ,sub = paste(&quot;GO Process:&quot;,tT[x,&quot;GO.Process&quot;]))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<pre class="r"><code>x = 3
#instead of having to look up IDs, just take the xth value of the TopTable...
plot(exprs(gset)[toString(tT[x,1]),] ~ fl #plot expressions for xth most differentially expressed gene
     ,main = paste(tT[x,&quot;Gene.symbol&quot;] , &quot; (#&quot;, x, &quot; most significantly differentially expressed gene)&quot;)
     ,xlab=&quot;Groups: G0 (Control), G1 (Chronic), G2 (Accelerated), G3 (Crises)&quot;,ylab=&quot;&quot;
     ,sub = paste(&quot;GO Process:&quot;,tT[x,&quot;GO.Process&quot;]))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<pre class="r"><code>x = 20
#instead of having to look up IDs, just take the xth value of the TopTable...
plot(exprs(gset)[toString(tT[x,1]),] ~ fl #plot expressions for xth most differentially expressed gene
     ,main = paste(tT[x,&quot;Gene.symbol&quot;] , &quot; (#&quot;, x, &quot; most significantly differentially expressed gene)&quot;)
     ,xlab=&quot;Groups: G0 (Control), G1 (Chronic), G2 (Accelerated), G3 (Crises)&quot;,ylab=&quot;&quot;
     ,sub = paste(&quot;GO Process:&quot;,tT[x,&quot;GO.Process&quot;]))</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<pre class="r"><code>#CAMP, id 10012670444
plot(exprs(gset)[&quot;10012670444&quot;,] ~ fl) #plot expressions for xth most differentially expressed gene</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
</div>
<div id="sidenote-what-if-we-build-a-different-contrast-model-what-if-we-do-non-pairwise-contrasts" class="section level3">
<h3>Sidenote: What if we build a different contrast model? What if we do non pairwise contrasts?</h3>
<div id="when-making-only-3-pairwise-contrastsg1-g0-g2-g0-g3-g0" class="section level5">
<h5>When making only 3 pairwise contrasts:G1-G0, G2-G0, G3-G0…</h5>
<p>The top 3 returned into toptable are the same, but then some of the ordering is different. All of the statistics within the groups were unchanged, but it shows that “make contrasts -&gt; top table” pulling in top table is based on all the differences within make contrasts.</p>
<pre class="r"><code>#build a linear model... 
fit_test &lt;- lmFit(gset, design)</code></pre>
<pre><code>## Warning: Partial NA coefficients for 607 probe(s)</code></pre>
<pre class="r"><code>cont.matrix &lt;- makeContrasts(G1-G0, G2-G0, G3-G0, levels=design) 
fit2_test &lt;- contrasts.fit(fit_test, cont.matrix)
fit2_test &lt;- eBayes(fit2_test, 0.01)
tT_test &lt;- topTable(fit2_test, adjust=&quot;fdr&quot;, sort.by=&quot;B&quot;, number=250) #sort by the lods or B-statistic</code></pre>
<pre class="r"><code>subset(tT_test, select=c(&quot;P.Value&quot;,&quot;Gene.symbol&quot;
                         ,&quot;G1...G0&quot;,&quot;G2...G0&quot;,&quot;G3...G0&quot;,&quot;AveExpr&quot;))[1:20,]</code></pre>
<pre><code>##                  P.Value Gene.symbol    G1...G0      G2...G0     G3...G0
## 10012672278 4.326511e-32         GAK  0.4329386  0.233988095  0.08015417
## 10012667916 6.967740e-29       VDAC2 -0.5550561 -0.257616667 -0.03490917
## 10012685105 1.508012e-28    FLJ40142 -0.7656912 -0.257484615 -0.07686500
## 10012672247 1.616099e-28         FGR  1.2933648  1.091309524  0.44514314
## 10012684680 2.798083e-28       ITGB2  1.1088035  0.903050000  0.39609231
## 10012688029 2.934467e-28       ITGAM  1.6470175  1.517516667  0.61402417
## 10012674495 9.584159e-28       BCAT1 -0.9783939 -0.685314286 -0.11528250
## 10012685898 1.084776e-27        TCP1 -0.6470132 -0.473578571 -0.24238750
## 10012676178 1.636838e-27       CD109 -1.2332658 -0.468750000 -0.14862000
## 10012672593 3.100368e-27       SOCS2 -0.5935421 -0.009833333  0.42497667
## 10012673694 3.886184e-27       TFB2M -0.7091298 -0.391792857 -0.17498000
## 10012681110 5.329463e-27       BCAT1 -0.9297307 -0.647135714 -0.08711250
## 10012670381 5.891002e-27        AVEN -0.5644053 -0.369216667 -0.08593917
## 10012688416 6.731142e-27         HK3  1.6237976  1.383229487  0.48534583
## 10012672906 6.981297e-27      TCIRG1  0.8570658  0.614367949  0.30966083
## 10012685376 1.184364e-26       ABCA7  0.7088792  0.522623810  0.14045667
## 10012665559 1.320298e-26      ADRBK1  0.5804500  0.416783333  0.13066216
## 10012673073 1.795678e-26      TCIRG1  0.8500711  0.612337179  0.30499833
## 10012672263 3.606272e-26       RIOK1 -0.5926465 -0.296228571 -0.18592500
## 10012682071 4.593546e-26     SLC15A3  1.3998886  1.321328571  0.71454750
##                AveExpr
## 10012672278 -0.1765462
## 10012667916  0.2728940
## 10012685105  0.4904940
## 10012672247 -0.4590925
## 10012684680 -0.4152991
## 10012688029 -0.6054675
## 10012674495  0.3885803
## 10012685898  0.2370034
## 10012676178  0.5833282
## 10012672593  0.5400863
## 10012673694  0.3059231
## 10012681110  0.4317872
## 10012670381  0.3205393
## 10012688416 -0.6322835
## 10012672906 -0.2994078
## 10012685376 -0.2949905
## 10012665559 -0.2442879
## 10012673073 -0.2952802
## 10012672263  0.2479145
## 10012682071 -0.3445556</code></pre>
</div>
<div id="combining-all-disease-state-samples-and-contrasting-with-the-control-again-the-top-20-genes-are-roughly-unchanged" class="section level5">
<h5>Combining all disease-state samples and contrasting with the control, again the top 20 genes are roughly unchanged:</h5>
<pre class="r"><code>fit_test &lt;- lmFit(gset, design)</code></pre>
<pre><code>## Warning: Partial NA coefficients for 607 probe(s)</code></pre>
<pre class="r"><code>cont.matrix &lt;- makeContrasts((G1+G2+G3)/3-G0, G1-G0, levels=design) 
fit2_test &lt;- contrasts.fit(fit_test, cont.matrix)
fit2_test &lt;- eBayes(fit2_test, 0.01)
tT_test &lt;- topTable(fit2_test, adjust=&quot;fdr&quot;, sort.by=&quot;B&quot;, number=250) #sort by the lods or B-statistic</code></pre>
<pre class="r"><code>subset(tT_test, select=c(&quot;P.Value&quot;,&quot;Gene.symbol&quot;
                         ,&quot;X.G1...G2...G3..3...G0&quot;,&quot;G1...G0&quot;,&quot;AveExpr&quot;))[1:20,]</code></pre>
<pre><code>##                  P.Value Gene.symbol X.G1...G2...G3..3...G0    G1...G0
## 10012672278 7.009431e-27         GAK             0.24902695  0.4329386
## 10012685105 2.280793e-25    FLJ40142            -0.36668028 -0.7656912
## 10012673381 1.990428e-24       TTC31             0.07240035  0.1806114
## 10012676178 4.205735e-24       CD109            -0.61687860 -1.2332658
## 10012672263 1.020676e-23       RIOK1            -0.35826669 -0.5926465
## 10012667916 1.197542e-23       VDAC2            -0.28252732 -0.5550561
## 10012678578 1.965737e-23        GFM1            -0.35646198 -0.7260904
## 10012673694 4.418057e-23       TFB2M            -0.42530089 -0.7091298
## 10012667702 7.888321e-23      TXNRD1            -0.31879403 -0.4976395
## 10012669509 8.166305e-23       ADAM6             0.14846311  0.3373912
## 10012685898 9.768178e-23        TCP1            -0.45432641 -0.6470132
## 10012672906 3.909375e-22      TCIRG1             0.59369819  0.8570658
## 10012673073 1.102750e-21      TCIRG1             0.58913552  0.8500711
## 10012684680 1.398506e-21       ITGB2             0.80264861  1.1088035
## 10012689315 1.737678e-21      PCDHB6             0.11713431  0.2814491
## 10012672247 3.609727e-21         FGR             0.94327248  1.2933648
## 10012672593 3.968896e-21       SOCS2            -0.05946626 -0.5935421
## 10012685751 8.146597e-21       TEX10            -0.37802369 -0.5982339
## 10012681527 8.810625e-21       NR5A1             0.25118615  0.4508474
## 10012667708 1.542147e-20      TYROBP             1.17904525  1.4830149
##                 AveExpr
## 10012672278 -0.17654615
## 10012685105  0.49049397
## 10012673381 -0.08940769
## 10012676178  0.58332821
## 10012672263  0.24791453
## 10012667916  0.27289402
## 10012678578  0.35475259
## 10012673694  0.30592308
## 10012667702  0.11465983
## 10012669509 -0.14166410
## 10012685898  0.23700342
## 10012672906 -0.29940776
## 10012673073 -0.29528017
## 10012684680 -0.41529914
## 10012689315 -0.13254359
## 10012672247 -0.45909252
## 10012672593  0.54008632
## 10012685751  0.24566207
## 10012681527 -0.23437179
## 10012667708 -0.50548462</code></pre>
</div>
<div id="why-not-follow-radich-lead-compare-chronic-with-acceleratedblast-crisis-but-also-subtract-out-healthy-cd34-profile-from-the-disease-samples." class="section level5">
<h5>Why not follow Radich’ lead: compare chronic with accelerated+blast crisis? But also subtract out healthy CD34 profile from the disease samples.</h5>
<p>Could be a good future project.</p>
</div>
</div>
<div id="ok-now-lets-try-some-enrichment" class="section level3">
<h3>Ok, now lets try some enrichment…</h3>
<pre class="r"><code>library(clusterProfiler)</code></pre>
<pre><code>## </code></pre>
<pre><code>## clusterProfiler v3.10.1  For help: https://guangchuangyu.github.io/software/clusterProfiler
## 
## If you use clusterProfiler in published research, please cite:
## Guangchuang Yu, Li-Gen Wang, Yanyan Han, Qing-Yu He. clusterProfiler: an R package for comparing biological themes among gene clusters. OMICS: A Journal of Integrative Biology. 2012, 16(5):284-287.</code></pre>
<pre class="r"><code>library(org.Hs.eg.db) </code></pre>
<pre><code>## Loading required package: AnnotationDbi</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:base&#39;:
## 
##     expand.grid</code></pre>
<pre><code>## </code></pre>
<pre class="r"><code>library(DOSE)</code></pre>
<pre><code>## DOSE v3.8.2  For help: https://guangchuangyu.github.io/DOSE
## 
## If you use DOSE in published research, please cite:
## Guangchuang Yu, Li-Gen Wang, Guang-Rong Yan, Qing-Yu He. DOSE: an R/Bioconductor package for Disease Ontology Semantic and Enrichment analysis. Bioinformatics 2015, 31(4):608-609</code></pre>
<div id="wrangle-the-data-by-finding-average-expression-for-3-of-the-phenotypes" class="section level4">
<h4>Wrangle the data by finding average expression for 3 of the phenotypes:</h4>
<pre class="r"><code>#exprs(gset)
#gset@phenoData@data
#gset$description
##dim(gset@assayData[[&quot;exprs&quot;]]) #24481 x 117

g0.means &lt;- rowMeans(exprs(gset)[,which(sml==&quot;G0&quot;)])
g1.means &lt;- rowMeans(exprs(gset)[,which(sml==&quot;G1&quot;)])
g3.means &lt;- rowMeans(exprs(gset)[,which(sml==&quot;G3&quot;)])</code></pre>
<pre class="r"><code>#create gene list
gene.list.3.1 &lt;- 10^g3.means / 10^g1.means #raw values are all log10 transormed. undo that.
#gene.list.3.1 &lt;- sort(gene.list.3.1, decreasing=TRUE)
gene.list.3.1 &lt;- log(gene.list.3.1, base=2)


gene.list.1.0 &lt;- 10^g1.means / 10^g0.means #raw values are all log10 transormed. undo that.
gene.list.1.0 &lt;- log(gene.list.1.0, base=2)</code></pre>
<pre class="r"><code>#before passing into the enrichment functions, need to translate proprietary (platform) ids into Entrez gene ids
platform.ids &lt;- names(gene.list.1.0)
#platform.ids
#platform.ids[1:30]

test.gene.list &lt;- gene.list.1.0[gset@featureData@data[,&quot;Gene.ID&quot;] != &quot;&quot;]
test.gene.list[1:10]</code></pre>
<pre><code>## 10012665102 10012665103 10012665105 10012665106 10012665108 10012665110 
##  -1.3124705  -0.5248442  -1.6525718  -1.6737447  -0.2690325   0.4636537 
## 10012665114 10012665116 10012665117 10012665118 
##  -1.0782425          NA  -0.4167504   0.6224128</code></pre>
<pre class="r"><code>test.gene.names &lt;- gset@featureData@data[,c(&quot;ID&quot;,&quot;Gene.ID&quot;)][gset@featureData@data[,&quot;Gene.ID&quot;] != &quot;&quot;,2]
test.gene.names[1:10]</code></pre>
<pre><code>##  [1] &quot;64388&quot;  &quot;140883&quot; &quot;2244&quot;   &quot;286133&quot; &quot;5879&quot;   &quot;55105&quot;  &quot;2823&quot;  
##  [8] &quot;399967&quot; &quot;124565&quot; &quot;150365&quot;</code></pre>
<pre class="r"><code>## feature 1: numeric vector
test = test.gene.list
## feature 2: named vector
names(test) = test.gene.names
## feature 3: decreasing order
test = sort(test, decreasing = TRUE)

#gset@featureData@data[,&quot;Gene.ID&quot;] != &quot;&quot;
#which(apply(gset@featureData@data[,&quot;Gene.ID&quot;],1,length) &gt; 0)[1:30]
test[1:30]</code></pre>
<pre><code>##     3684     2219      934    79887     6283     2215     9536   144423 
## 5.471274 5.317501 5.235324 5.211114 5.170814 5.079044 5.070349 5.038089 
##      383     7305     1536    10288     8291     2357    26030     3934 
## 4.941359 4.926469 4.878348 4.805924 4.762858 4.740333 4.737145 4.729680 
##     8875     3240     8807    26034      240    51296     8218   199675 
## 4.721287 4.715599 4.713624 4.684761 4.665555 4.650329 4.645480 4.641555 
##     1087      306    11240      929     3579     2220 
## 4.629955 4.613814 4.538439 4.512597 4.451375 4.439235</code></pre>
<p>so there were a lot of NA rows in the genelist i created. 3k of 18k had to be removed because they were na! Was messing with the enrichment functions.</p>
<pre class="r"><code>test &lt;- test[!is.na(test)]
#test[15250:15500] #what is 100128545 doing here??? ...gues it&#39;s a legit gene ID
#uhh.... I should get rid of all of these NA values!!!
#good source: https://github.com/GuangchuangYu/DOSE/wiki/how-to-prepare-your-own-geneList</code></pre>
<p>Distribution of fold change, chronic phase vs healthy control</p>
<pre class="r"><code>plot(test)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
</div>
<div id="go-enrichment-analysis-molecular-function-level-2-chronic-phase-vs-healthy-control" class="section level4">
<h4>GO Enrichment analysis, Molecular function (level 2), chronic phase vs healthy control</h4>
<pre class="r"><code>gene &lt;- names(test)[abs(test) &gt; 2]
ggo &lt;- groupGO(gene     = gene,
               OrgDb    = org.Hs.eg.db,
               ont      = &quot;MF&quot;,
               level    = 2,
               readable = TRUE)
#order by count
ggo@result = ggo@result[order(ggo@result[,&quot;Count&quot;],decreasing=TRUE),]
ggo@result[1:12,1:4]</code></pre>
<pre><code>##                    ID                      Description Count GeneRatio
## GO:0005488 GO:0005488                          binding   743   743/912
## GO:0003824 GO:0003824               catalytic activity   288   288/912
## GO:0098772 GO:0098772     molecular function regulator   123   123/912
## GO:0060089 GO:0060089    molecular transducer activity   104   104/912
## GO:0140110 GO:0140110 transcription regulator activity    92    92/912
## GO:0005215 GO:0005215             transporter activity    64    64/912
## GO:0005198 GO:0005198     structural molecule activity    37    37/912
## GO:0104005 GO:0104005      hijacked molecular function     9     9/912
## GO:0038024 GO:0038024          cargo receptor activity     8     8/912
## GO:0016209 GO:0016209             antioxidant activity     6     6/912
## GO:0140104 GO:0140104       molecular carrier activity     3     3/912
## GO:0045182 GO:0045182   translation regulator activity     2     2/912</code></pre>
<pre class="r"><code>barplot(ggo,showCategory=8)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
</div>
<div id="go-enrichment-analysis-molecular-function-level-3-chronic-phase-vs-healthy-control" class="section level4">
<h4>GO Enrichment analysis, Molecular function (level 3), chronic phase vs healthy control</h4>
<pre class="r"><code>gene &lt;- names(test)[abs(test) &gt; 2]
ggo &lt;- groupGO(gene     = gene,
               OrgDb    = org.Hs.eg.db,
               ont      = &quot;MF&quot;,
               level    = 3,
               readable = TRUE)
#order by count
ggo@result = ggo@result[order(ggo@result[,&quot;Count&quot;],decreasing=TRUE),]
ggo@result[1:12,2:4]</code></pre>
<pre><code>##                                          Description Count GeneRatio
## GO:0005515                           protein binding   623   623/912
## GO:0043167                               ion binding   314   314/912
## GO:0097159           organic cyclic compound binding   249   249/912
## GO:1901363             heterocyclic compound binding   244   244/912
## GO:0016787                        hydrolase activity   131   131/912
## GO:0036094                    small molecule binding   130   130/912
## GO:0097367           carbohydrate derivative binding   121   121/912
## GO:0016740                      transferase activity   119   119/912
## GO:0140096   catalytic activity, acting on a protein   102   102/912
## GO:0038023               signaling receptor activity    97    97/912
## GO:0008144                              drug binding    90    90/912
## GO:0003700 DNA-binding transcription factor activity    75    75/912</code></pre>
<pre class="r"><code>barplot(ggo,showCategory=12)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
</div>
<div id="go-enrichment-analysis-biologic-process-level-3-chronic-phase-vs-healthy-control" class="section level4">
<h4>GO Enrichment analysis, Biologic Process (level 3), chronic phase vs healthy control</h4>
<pre class="r"><code>gene &lt;- names(test)[abs(test) &gt; 2]
ggo &lt;- groupGO(gene     = gene,
               OrgDb    = org.Hs.eg.db,
               ont      = &quot;BP&quot;,
               level    = 3,
               readable = TRUE)

#order by count
ggo@result = ggo@result[order(ggo@result[,&quot;Count&quot;],decreasing=TRUE),]
ggo@result[1:12,2:4]</code></pre>
<pre><code>##                                          Description Count GeneRatio
## GO:0050789          regulation of biological process   625   625/912
## GO:0050794            regulation of cellular process   589   589/912
## GO:0071704       organic substance metabolic process   534   534/912
## GO:0044237                cellular metabolic process   521   521/912
## GO:0044238                 primary metabolic process   512   512/912
## GO:0006807       nitrogen compound metabolic process   492   492/912
## GO:0051716             cellular response to stimulus   449   449/912
## GO:0007154                        cell communication   407   407/912
## GO:0007165                       signal transduction   385   385/912
## GO:0048518 positive regulation of biological process   373   373/912
## GO:0048856          anatomical structure development   351   351/912
## GO:0019222           regulation of metabolic process   349   349/912</code></pre>
<pre class="r"><code>barplot(ggo,showCategory=12)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-31-1.png" width="672" /></p>
</div>
<div id="kegg-enrichment-analysis-chronic-phase-vs-healthy-control" class="section level4">
<h4>KEGG Enrichment analysis, chronic phase vs healthy control</h4>
<ul>
<li>Default p value cutoff (5%)</li>
<li>Further information about enirchment scores: <a href="https://yulab-smu.github.io/clusterProfiler-book/chapter2.html#gene-set-enrichment-analysis" class="uri">https://yulab-smu.github.io/clusterProfiler-book/chapter2.html#gene-set-enrichment-analysis</a></li>
</ul>
<pre class="r"><code>#KEGG Gene Set Enrichment Analysis
kk2 &lt;- gseKEGG(geneList     = test,
               organism     = &#39;hsa&#39;,
               nPerm        = 1000,
               minGSSize    = 100,
               pvalueCutoff = 0.05,
               verbose      = FALSE)</code></pre>
<pre><code>## Warning in fgsea(pathways = geneSets, stats = geneList, nperm = nPerm, minSize = minGSSize, : There are ties in the preranked stats (0.4% of the list).
## The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<pre><code>## Warning in fgsea(pathways = geneSets, stats = geneList, nperm = nPerm,
## minSize = minGSSize, : There are duplicate gene names, fgsea may produce
## unexpected results</code></pre>
<pre class="r"><code>kk2@result = kk2@result[order(kk2@result[,&quot;setSize&quot;],decreasing=TRUE),]
kk2@result[1:12,2:7]</code></pre>
<pre><code>##                                       Description setSize enrichmentScore
## hsa05200                       Pathways in cancer     458       0.2403397
## hsa05165           Human papillomavirus infection     267       0.3156233
## hsa04080  Neuroactive ligand-receptor interaction     258       0.3083987
## hsa04060   Cytokine-cytokine receptor interaction     232       0.4055668
## hsa04144                              Endocytosis     201       0.4053610
## hsa04014                    Ras signaling pathway     194       0.2572864
## hsa05166  Human T-cell leukemia virus 1 infection     194       0.2666884
## hsa05163          Human cytomegalovirus infection     191       0.3905283
## hsa04015                   Rap1 signaling pathway     180       0.3437283
## hsa05169             Epstein-Barr virus infection     173       0.2579447
## hsa04810         Regulation of actin cytoskeleton     172       0.3567105
## hsa05170 Human immunodeficiency virus 1 infection     172       0.3780215
##               NES      pvalue   p.adjust
## hsa05200 1.343580 0.022222222 0.04537720
## hsa05165 1.695539 0.009523810 0.02380952
## hsa04080 1.642607 0.009523810 0.02380952
## hsa04060 2.124492 0.008333333 0.02222222
## hsa04144 2.101513 0.007575758 0.02181025
## hsa04014 1.321299 0.015267176 0.03301011
## hsa05166 1.369582 0.015267176 0.03301011
## hsa05163 1.997701 0.007633588 0.02181025
## hsa04015 1.739556 0.008064516 0.02222222
## hsa05169 1.313993 0.021428571 0.04511278
## hsa04810 1.822859 0.006896552 0.02122016
## hsa05170 1.931762 0.006896552 0.02122016</code></pre>
<pre class="r"><code>plot(kk2@result[,&quot;setSize&quot;])</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-33-1.png" width="672" /></p>
</div>
<div id="kegg-over-representation-analysis-chronic-phase-vs-healthy-control" class="section level4">
<h4>KEGG Over Representation analysis, chronic phase vs healthy control</h4>
<pre class="r"><code>#KEGG over representation Analysis
kk3 &lt;- enrichKEGG(gene    = names(test),
               organism     = &#39;hsa&#39;,
               pvalueCutoff = 0.05
  )
kk3@result[1:12,c(&quot;Description&quot;,&quot;GeneRatio&quot;,&quot;p.adjust&quot;,&quot;Count&quot;)]</code></pre>
<pre><code>##                                      Description GeneRatio     p.adjust
## hsa04110                              Cell cycle  117/5603 1.519569e-08
## hsa04151              PI3K-Akt signaling pathway  303/5603 1.519569e-08
## hsa05166 Human T-cell leukemia virus 1 infection  194/5603 5.227902e-08
## hsa04010                  MAPK signaling pathway  254/5603 7.468306e-08
## hsa04015                  Rap1 signaling pathway  180/5603 1.615953e-06
## hsa05202 Transcriptional misregulation in cancer  163/5603 3.944984e-06
## hsa04520                       Adherens junction   69/5603 5.123327e-06
## hsa04640              Hematopoietic cell lineage   90/5603 5.797798e-06
## hsa05152                            Tuberculosis  156/5603 1.104496e-05
## hsa05212                       Pancreatic cancer   71/5603 1.104496e-05
## hsa05161                             Hepatitis B  143/5603 1.188241e-05
## hsa05169            Epstein-Barr virus infection  173/5603 1.354172e-05
##          Count
## hsa04110   117
## hsa04151   303
## hsa05166   194
## hsa04010   254
## hsa04015   180
## hsa05202   163
## hsa04520    69
## hsa04640    90
## hsa05152   156
## hsa05212    71
## hsa05161   143
## hsa05169   173</code></pre>
<pre class="r"><code>library(GSEABase)</code></pre>
<pre><code>## Loading required package: annotate</code></pre>
<pre><code>## Loading required package: XML</code></pre>
<pre><code>## Loading required package: graph</code></pre>
<pre><code>## 
## Attaching package: &#39;graph&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:XML&#39;:
## 
##     addNode</code></pre>
<pre class="r"><code>library(tidyr)</code></pre>
<pre><code>## 
## Attaching package: &#39;tidyr&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:S4Vectors&#39;:
## 
##     expand</code></pre>
</div>
<div id="wikipathways-enrichment-analysis-chronic-phase-vs.-healthy-control" class="section level4">
<h4>Wikipathways enrichment analysis, Chronic phase vs. healthy control</h4>
<pre class="r"><code>gene &lt;- names(test)[abs(test) &gt; 2]

wpgmtfile &lt;- system.file(&quot;extdata&quot;, &quot;wikipathways-20180810-gmt-Homo_sapiens.gmt&quot;, package=&quot;clusterProfiler&quot;)
wp2gene &lt;- read.gmt(wpgmtfile) #i had to install GSEAbase first ??
wp2gene &lt;- wp2gene %&gt;% tidyr::separate(ont, c(&quot;name&quot;,&quot;version&quot;,&quot;wpid&quot;,&quot;org&quot;), &quot;%&quot;)
wpid2gene &lt;- wp2gene %&gt;% dplyr::select(wpid, gene) #TERM2GENE
wpid2name &lt;- wp2gene %&gt;% dplyr::select(wpid, name) #TERM2NAME

ewp &lt;- enricher(gene, TERM2GENE = wpid2gene, TERM2NAME = wpid2name)
ewp &lt;- setReadable(ewp, org.Hs.eg.db, keyType = &quot;ENTREZID&quot;)
ewp@result = ewp@result[order(ewp@result[,&quot;Count&quot;],decreasing=TRUE),]
ewp@result[1:12,2:6]</code></pre>
<pre><code>##                                           Description GeneRatio  BgRatio
## WP2882                 Nuclear Receptors Meta-Pathway    28/388 321/6249
## WP4172                     PI3K-Akt Signaling Pathway    23/388 345/6249
## WP3888                 VEGFA-VEGFR2 Signaling Pathway    21/388 238/6249
## WP3932 Focal Adhesion-PI3K-Akt-mTOR-signaling pathway    21/388 309/6249
## WP2431                             Spinal Cord Injury    20/388 120/6249
## WP2806                        Human Complement System    17/388  98/6249
## WP2877                     Vitamin D Receptor Pathway    17/388 186/6249
## WP382                          MAPK Signaling Pathway    17/388 249/6249
## WP455                   GPCRs, Class A Rhodopsin-like    17/388 258/6249
## WP3945                          TYROBP Causal Network    16/388  63/6249
## WP289  Myometrial Relaxation and Contraction Pathways    16/388 158/6249
## WP536          Calcium Regulation in the Cardiac Cell    15/388 151/6249
##              pvalue    p.adjust
## WP2882 4.095917e-02 0.769103011
## WP4172 3.914563e-01 0.991674283
## WP3888 6.380244e-02 0.769103011
## WP3932 3.645954e-01 0.980635778
## WP2431 4.103470e-05 0.004267609
## WP2806 9.260321e-05 0.007223050
## WP2877 6.906871e-02 0.769103011
## WP382  3.776371e-01 0.983076082
## WP455  4.358641e-01 0.998193743
## WP3945 8.843877e-07 0.000275929
## WP289  3.540420e-02 0.769103011
## WP536  4.706966e-02 0.769103011</code></pre>
<pre class="r"><code>barplot(ewp,showCategory=12)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
<pre class="r"><code>ewp2 &lt;- GSEA(test, TERM2GENE = wpid2gene, TERM2NAME = wpid2name, verbose=FALSE)</code></pre>
<pre><code>## Warning in fgsea(pathways = geneSets, stats = geneList, nperm = nPerm, minSize = minGSSize, : There are ties in the preranked stats (0.4% of the list).
## The order of those tied genes will be arbitrary, which may produce unexpected results.</code></pre>
<pre><code>## Warning in fgsea(pathways = geneSets, stats = geneList, nperm = nPerm,
## minSize = minGSSize, : There are duplicate gene names, fgsea may produce
## unexpected results</code></pre>
<pre class="r"><code>ewp2 &lt;- setReadable(ewp2, org.Hs.eg.db, keyType = &quot;ENTREZID&quot;)
ewp2@result = ewp2@result[order(ewp2@result[,&quot;setSize&quot;],decreasing=TRUE),]
ewp2@result</code></pre>
<pre><code>##            ID
## WP382   WP382
## WP455   WP455
## WP306   WP306
## WP4223 WP4223
## WP437   WP437
## WP3929 WP3929
## WP2877 WP2877
## WP536   WP536
## WP2380 WP2380
## WP51     WP51
## WP1449 WP1449
## WP411   WP411
## WP179   WP179
## WP3915 WP3915
## WP4217 WP4217
## WP2431 WP2431
## WP710   WP710
## WP185   WP185
## WP75     WP75
## WP23     WP23
## WP2806 WP2806
## WP231   WP231
## WP4263 WP4263
## WP2446 WP2446
## WP2328 WP2328
## WP69     WP69
## WP2059 WP2059
## WP4298 WP4298
## WP2037 WP2037
## WP2324 WP2324
## WP24     WP24
## WP3863 WP3863
## WP45     WP45
## WP304   WP304
## WP195   WP195
## WP3945 WP3945
## WP395   WP395
## WP585   WP585
## WP4205 WP4205
## WP2637 WP2637
## WP268   WP268
## WP286   WP286
## WP466   WP466
## WP3941 WP3941
## WP107   WP107
## WP1591 WP1591
## WP364   WP364
## WP2848 WP2848
## WP2406 WP2406
## WP383   WP383
## WP49     WP49
## WP4136 WP4136
## WP127   WP127
## WP3937 WP3937
## WP2525 WP2525
## WP313   WP313
## WP453   WP453
## WP422   WP422
## WP581   WP581
## WP4197 WP4197
## WP4301 WP4301
## WP3678 WP3678
##                                                                         Description
## WP382                                                        MAPK Signaling Pathway
## WP455                                                 GPCRs, Class A Rhodopsin-like
## WP306                                                                Focal Adhesion
## WP4223                                                                Ras Signaling
## WP437                                                    EGF/EGFR Signaling Pathway
## WP3929                                                  Chemokine signaling pathway
## WP2877                                                   Vitamin D Receptor Pathway
## WP536                                        Calcium Regulation in the Cardiac Cell
## WP2380                   Brain-Derived Neurotrophic Factor (BDNF) signaling pathway
## WP51                                               Regulation of Actin Cytoskeleton
## WP1449                           Regulation of toll-like receptor signaling pathway
## WP411                                                               mRNA Processing
## WP179                                                                    Cell Cycle
## WP3915                               Angiopoietin Like Protein 8 Regulatory Pathway
## WP4217                                                  Ebola Virus Pathway on Host
## WP2431                                                           Spinal Cord Injury
## WP710                                      DNA Damage Response (only ATM dependent)
## WP185                                               Integrin-mediated Cell Adhesion
## WP75                                           Toll-like Receptor Signaling Pathway
## WP23                                              B Cell Receptor Signaling Pathway
## WP2806                                                      Human Complement System
## WP231                                                   TNF alpha Signaling Pathway
## WP4263                                            Pancreatic adenocarcinoma pathway
## WP2446                                                Retinoblastoma (RB) in Cancer
## WP2328                                                          Allograft Rejection
## WP69                               T-Cell antigen Receptor (TCR)  Signaling Pathway
## WP2059                                                           Alzheimers Disease
## WP4298                                                      Viral Acute Myocarditis
## WP2037                                                  Prolactin Signaling Pathway
## WP2324                                                             AGE/RAGE pathway
## WP24                                                                  Peptide GPCRs
## WP3863 T-Cell antigen Receptor (TCR) pathway during Staphylococcus aureus infection
## WP45                                                     G1 to S cell cycle control
## WP304                                                Kit receptor signaling pathway
## WP195                                                        IL-1 signaling pathway
## WP3945                                                        TYROBP Causal Network
## WP395                                                        IL-4 Signaling Pathway
## WP585                                          Interferon type I signaling pathways
## WP4205                                 MET in type 1 papillary renal cell carcinoma
## WP2637                                   Structural Pathway of Interleukin 1 (IL-1)
## WP268                                                       Notch Signaling Pathway
## WP286                                                        IL-3 Signaling Pathway
## WP466                                                               DNA Replication
## WP3941                                                             Oxidative Damage
## WP107                                                           Translation Factors
## WP1591                                                            Heart Development
## WP364                                                        IL-6 signaling pathway
## WP2848                                                      Differentiation Pathway
## WP2406                                           Cardiac Progenitor Differentiation
## WP383                                                   Striated Muscle Contraction
## WP49                                                         IL-2 Signaling Pathway
## WP4136                               Fibrin Complement Receptor 3 Signaling Pathway
## WP127                                                        IL-5 Signaling Pathway
## WP3937                                      Microglia Pathogen Phagocytosis Pathway
## WP2525                                  Trans-sulfuration and one carbon metabolism
## WP313                                Signaling of Hepatocyte Growth Factor Receptor
## WP453                                                 Inflammatory Response Pathway
## WP422                                                                  MAPK Cascade
## WP581                                                        EPO Receptor Signaling
## WP4197                                    The human immune response to tuberculosis
## WP4301  Inhibition of exosome biogenesis and secretion by Manumycin A in CRPC cells
## WP3678       Amplification and Expansion of Oncogenic Pathways as Metastatic Traits
##        setSize enrichmentScore       NES      pvalue   p.adjust    qvalues
## WP382      208       0.3178966  1.665844 0.006896552 0.04249224 0.03203880
## WP455      178       0.4512969  2.308863 0.006493506 0.04249224 0.03203880
## WP306      170       0.2985419  1.508665 0.006369427 0.04249224 0.03203880
## WP4223     152       0.2990082  1.514315 0.005617978 0.04003599 0.03018680
## WP437      140       0.3059201  1.508735 0.005847953 0.04003599 0.03018680
## WP3929     137       0.4774793  2.337999 0.005813953 0.04003599 0.03018680
## WP2877     136       0.3565217  1.744035 0.005714286 0.04003599 0.03018680
## WP536      129       0.3195664  1.543505 0.005434783 0.04003599 0.03018680
## WP2380     126       0.3177768  1.530624 0.005347594 0.04003599 0.03018680
## WP51       125       0.3563179  1.716552 0.005376344 0.04003599 0.03018680
## WP1449     120       0.3817597  1.820713 0.005434783 0.04003599 0.03018680
## WP411      115      -0.3816842 -1.547045 0.004944376 0.04003599 0.03018680
## WP179      113      -0.4117301 -1.665464 0.001236094 0.04003599 0.03018680
## WP3915     113       0.3634181  1.722266 0.005181347 0.04003599 0.03018680
## WP4217     106       0.3768360  1.764218 0.004950495 0.04003599 0.03018680
## WP2431      97       0.3238284  1.507694 0.004651163 0.04003599 0.03018680
## WP710       92       0.3440662  1.583859 0.004504505 0.04003599 0.03018680
## WP185       90       0.4790033  2.201506 0.004484305 0.04003599 0.03018680
## WP75        88       0.4060789  1.857580 0.004385965 0.04003599 0.03018680
## WP23        83       0.4410884  2.008138 0.004237288 0.04003599 0.03018680
## WP2806      83       0.5182619  2.359485 0.004237288 0.04003599 0.03018680
## WP231       82       0.4006853  1.796738 0.004566210 0.04003599 0.03018680
## WP4263      82       0.3373105  1.512556 0.004566210 0.04003599 0.03018680
## WP2446      81      -0.5073828 -1.960711 0.001283697 0.04003599 0.03018680
## WP2328      80       0.4041253  1.802870 0.004504505 0.04003599 0.03018680
## WP69        78       0.3924976  1.741897 0.004385965 0.04003599 0.03018680
## WP2059      74       0.3587712  1.573204 0.004310345 0.04003599 0.03018680
## WP4298      74       0.3660628  1.605178 0.004310345 0.04003599 0.03018680
## WP2037      67       0.4416420  1.929609 0.004081633 0.04003599 0.03018680
## WP2324      61       0.4012882  1.727845 0.003831418 0.04003599 0.03018680
## WP24        60       0.5659110  2.442196 0.003816794 0.04003599 0.03018680
## WP3863      56       0.4583000  1.943378 0.003745318 0.04003599 0.03018680
## WP45        55      -0.5003720 -1.808360 0.001360544 0.04003599 0.03018680
## WP304       52       0.3818668  1.585986 0.007326007 0.04307059 0.03247487
## WP195       51       0.4200820  1.721897 0.003703704 0.04003599 0.03018680
## WP3945      50       0.6596732  2.692508 0.003663004 0.04003599 0.03018680
## WP395       50       0.4994371  2.038492 0.003663004 0.04003599 0.03018680
## WP585       49       0.4098614  1.659942 0.007380074 0.04307059 0.03247487
## WP4205      46       0.4305781  1.720719 0.003676471 0.04003599 0.03018680
## WP2637      45       0.4184183  1.660860 0.003649635 0.04003599 0.03018680
## WP268       41       0.4173131  1.645489 0.007042254 0.04249224 0.03203880
## WP286       40       0.6490329  2.573594 0.003412969 0.04003599 0.03018680
## WP466       40      -0.5106191 -1.721158 0.004231312 0.04003599 0.03018680
## WP3941      39       0.4289240  1.680469 0.003448276 0.04003599 0.03018680
## WP107       39      -0.5045288 -1.698100 0.007022472 0.04249224 0.03203880
## WP1591      38       0.4326291  1.684146 0.003424658 0.04003599 0.03018680
## WP364       38       0.5013003  1.951470 0.003424658 0.04003599 0.03018680
## WP2848      38       0.4233857  1.648163 0.006849315 0.04249224 0.03203880
## WP2406      37       0.4441464  1.706797 0.003460208 0.04003599 0.03018680
## WP383       34       0.5092571  1.895862 0.003378378 0.04003599 0.03018680
## WP49        34       0.4953567  1.844113 0.003378378 0.04003599 0.03018680
## WP4136      33       0.4839941  1.785571 0.003389831 0.04003599 0.03018680
## WP127       33       0.4525469  1.669555 0.006779661 0.04249224 0.03203880
## WP3937      32       0.8055767  2.934957 0.003424658 0.04003599 0.03018680
## WP2525      31      -0.5378836 -1.751713 0.005641749 0.04003599 0.03018680
## WP313       28       0.5052868  1.789384 0.003267974 0.04003599 0.03018680
## WP453       28       0.5782790  2.047873 0.003267974 0.04003599 0.03018680
## WP422       26       0.6254933  2.183195 0.003076923 0.04003599 0.03018680
## WP581       24       0.5578240  1.907964 0.002985075 0.04003599 0.03018680
## WP4197      19       0.5812384  1.883726 0.008645533 0.04964209 0.03742973
## WP4301      14       0.6605187  1.911172 0.005763689 0.04003599 0.03018680
## WP3678      13       0.7172974  2.055907 0.002754821 0.04003599 0.03018680
##        rank                   leading_edge
## WP382  2377 tags=22%, list=15%, signal=20%
## WP455  3782 tags=44%, list=23%, signal=34%
## WP306  3890 tags=36%, list=24%, signal=28%
## WP4223 2787 tags=27%, list=17%, signal=23%
## WP437  2827 tags=29%, list=17%, signal=24%
## WP3929 3024 tags=35%, list=19%, signal=29%
## WP2877 1907 tags=24%, list=12%, signal=21%
## WP536  3107 tags=32%, list=19%, signal=26%
## WP2380 3024 tags=27%, list=19%, signal=22%
## WP51   2937 tags=32%, list=18%, signal=27%
## WP1449 3292 tags=30%, list=20%, signal=24%
## WP411  6967 tags=61%, list=43%, signal=35%
## WP179  4428 tags=44%, list=27%, signal=32%
## WP3915 3236 tags=32%, list=20%, signal=26%
## WP4217 2341 tags=23%, list=14%, signal=20%
## WP2431 3024 tags=31%, list=19%, signal=25%
## WP710  3024 tags=26%, list=19%, signal=21%
## WP185  3324 tags=42%, list=20%, signal=34%
## WP75   3147 tags=30%, list=19%, signal=25%
## WP23   2159 tags=31%, list=13%, signal=27%
## WP2806  874  tags=26%, list=5%, signal=25%
## WP231  2334 tags=30%, list=14%, signal=26%
## WP4263 2923 tags=26%, list=18%, signal=22%
## WP2446 5814 tags=67%, list=36%, signal=43%
## WP2328 2429 tags=29%, list=15%, signal=25%
## WP69   2159 tags=25%, list=13%, signal=22%
## WP2059 4288 tags=41%, list=26%, signal=30%
## WP4298 2905 tags=28%, list=18%, signal=23%
## WP2037 3003 tags=33%, list=19%, signal=27%
## WP2324 3943 tags=35%, list=24%, signal=27%
## WP24   3511 tags=51%, list=22%, signal=40%
## WP3863 2483 tags=36%, list=15%, signal=30%
## WP45   5791 tags=58%, list=36%, signal=38%
## WP304  2492 tags=27%, list=15%, signal=23%
## WP195  4184 tags=46%, list=26%, signal=34%
## WP3945 1638 tags=49%, list=10%, signal=44%
## WP395  2822 tags=37%, list=17%, signal=31%
## WP585  2492 tags=28%, list=15%, signal=24%
## WP4205 2765 tags=29%, list=17%, signal=24%
## WP2637 3089 tags=35%, list=19%, signal=28%
## WP268  3722 tags=49%, list=23%, signal=38%
## WP286  2492 tags=39%, list=15%, signal=33%
## WP466  5656 tags=70%, list=35%, signal=46%
## WP3941 1616 tags=21%, list=10%, signal=19%
## WP107  5715 tags=63%, list=35%, signal=41%
## WP1591 4036 tags=41%, list=25%, signal=31%
## WP364  4293 tags=45%, list=26%, signal=33%
## WP2848 3031 tags=33%, list=19%, signal=27%
## WP2406 2905 tags=44%, list=18%, signal=37%
## WP383  2616 tags=34%, list=16%, signal=29%
## WP49   4293 tags=53%, list=26%, signal=39%
## WP4136  292  tags=12%, list=2%, signal=12%
## WP127  4277 tags=47%, list=26%, signal=35%
## WP3937  820  tags=45%, list=5%, signal=43%
## WP2525 3707 tags=54%, list=23%, signal=42%
## WP313  2798 tags=31%, list=17%, signal=26%
## WP453  2639 tags=43%, list=16%, signal=36%
## WP422  2922 tags=53%, list=18%, signal=43%
## WP581  5369 tags=77%, list=33%, signal=51%
## WP4197 5369 tags=67%, list=33%, signal=45%
## WP4301 2765 tags=62%, list=17%, signal=52%
## WP3678 2452 tags=56%, list=15%, signal=48%
##                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      core_enrichment
## WP382                                                                                                                              CD14/LRRK2/LRRK2/CACNA1D/PRKCD/IL1R1/IL1R2/MAPK3/PAK1/FLNA/RAC2/ARRB2/FASLG/CACNA2D3/DUSP16/MAP3K1/FGFR4/MAX/FGF14/TNFRSF1A/TGFBR2/MAP4K4/MAP2K3/CACNB4/MAP2K5/PRKACA/MAP3K11/MAPK14/MAPK1/STK4/MAP3K1/MAPK7/TGFBR1/GNG12/MAPK13/CACNA1D/MAPK8IP2/MAP3K2/MOS/CACNA1B/IKBKG/STK4/MAP3K2/FGF17/RAF1/MAP2K6/RAC2/SOS1/KRAS/FGF5/CACNA1H/CACNA1I/AKT1/CACNA1A/CACNG2/RRAS/RASGRF2/PPP5C/AKT2/MAPK8IP1
## WP455  FPR1/CXCR2/FPR2/CXCR1/CCR1/ADORA2A/PTGDR/C3AR1/LTB4R/GPR68/OPRL1/GPR65/PTAFR/CCR3/C5AR2/CCR7/CXCR5/P2RY10/CCR6/HCAR3/ADRA2B/HTR1A/HTR6/HTR1B/AGTR2/P2RY6/ADORA3/GPR37/GPR17/P2RY2/MTNR1A/GPR15/NMBR/SSTR3/CCKAR/NPY1R/MC2R/CCKBR/HRH3/OR10H2/DRD2/HCRTR1/OR1D2/HTR2B/SSTR4/AGTR1/OPRL1/NPBWR2/GPR45/GPR3/APLNR/AVPR1B/GPR4/P2RY14/CHRM2/BDKRB1/HRH1/AVPR2/CHRM4/CNR1/OR10J1/NMUR1/ACKR4/GPR34/HTR1E/HRH1/GHSR/PTGFR/ADRA1A/AGTR2/GPER1/ADRB2/GALR1/OR10H1/MC1R/DRD5/SSTR2/GPR6/ADRA2A/MTNR1B/ADORA1/RRH/GPR1/OR1F1/NMUR2/MC5R
## WP306                                                              PXN/VASP/MAPK3/PAK1/VCL/PTEN/FLNA/RAC2/THBS4/VCL/COMP/BLK/ACTB/VEGFA/CCND3/PTK2/MAPK1/PPP1R12B/MYL9/ITGA3/LAMB3/COL4A2/RAPGEF1/ITGA1/VAV1/FYN/PDPK1/TLN1/TNK2/ITGA8/MYLPF/COL4A4/IBSP/ZYX/PRKCB/RAF1/RAC2/SOS1/AKT1/LAMB2/IGF1/RHOA/ITGB7/ITGA5/PPP1R12B/PIK3CD/PDGFRA/AKT2/TNC/ZYX/ILK/PTK6/PAK4/CAV3/PRKCG/VTN/MAPK1/PDPK1/COL2A1/GSK3B/SHC3/ITGA8/COL6A2/PPP1R12B/COL5A2/SHC1/PAK3/THBS3/TXK/SHC4/MAP2K1/PARVG/RHOA/COL4A1/ACTN4/RAC1/CAPN2/EGFR/TNN/PGF/LAMC3
## WP4223                                                                                                                                                                                               RASSF5/MAPK3/PAK1/RAC2/FASLG/GNB2/FGFR4/RASAL3/CALML4/PLCG2/PRKACA/CALM1/ETS1/GNGT1/CSF1R/MAPK1/RAB5C/STK4/RIN1/CALML4/GNG12/GNB1/GNG2/IKBKG/GNGT2/RALB/RGL2/PRKCB/STK4/BCL2L1/PLA2G2D/RAF1/GNG5/RAC2/SOS1/KRAS/CALML3/AKT1/RHOA/PLA2G5/KSR1/RRAS/RASGRF2/PIK3CD/PDGFRA/AKT2/GNG7/ZAP70/GNG2/PAK4/PRKCG/PLD1/RASAL2/MAPK1/TIAM1
## WP437                                                                                                                                                                                                 IQSEC1/PRKCD/PAK1/CBL/PTEN/MAP3K3/ARHGEF1/SH3KBP1/MAP3K1/CBL/RPS6KA2/IQGAP1/MAP2K5/REPS2/PLSCR1/MAPK14/PTK2/MAPK1/CBL/SYNJ1/MAP3K1/CFL1/RIN1/MAPK7/RPS6KA1/SH3KBP1/REPS2/VAV1/PDPK1/TNK2/MAP3K2/RALB/PRKCB/MAP3K2/RAF1/JAK1/NCOA3/MAP3K3/SOS1/KRAS/AKT1/INPP5D/STAT3/CSK/CAMK2A/PTK6/PLD1/HGS/RPS6KA2/MAPK1/PDPK1/PTK2B/PTPN12
## WP3929                                                                                                                                                                                  CXCR2/NCF1/CCR1/CCL5/PXN/ADCY4/PRKCD/WAS/PIK3CG/GRK6/MAPK3/PPBP/PAK1/RAC2/ARRB2/GNB2/CCR3/GNAI2/GRK5/LYN/CCR7/NFKBIB/CCL4/ITK/CXCR5/GNGT1/CCR6/PTK2/MAPK1/PLCB2/DOCK2/GNG12/VAV1/GNB1/GNG2/IKBKG/GNGT2/PRKCB/RAF1/GNG5/RAC2/SOS1/KRAS/AKT1/PIK3R5/PIK3CD/ADCY9/AKT2/GNG7/STAT3/GNG2/CXCR6/CSK/MAPK1/TIAM1/PTK2B/CCL17/GSK3B/SHC3/CCL19/FOXO3
## WP2877                                                                                                                                                                                                                                                                               ITGAM/ALOX5/CD14/TREM1/S100A6/BCL6/ORM2/G6PD/RXRA/STEAP4/CRACR2B/ATP2C2/VDR/LPGAT1/CDC34/CEACAM1/S100A4/CASP5/S100A4/MXD1/SLC8A1/MXD1/IL1RL1/ADAMTS5/NINJ1/ADAMTS5/THBD/PPARD/CLMN/S100G/ID4/CYP2D6/CYP1A1/MXD1/CEBPA/TNFRSF11B/EFNA5/CLMN/THBD
## WP536                                                                                                                                                                                                       CACNA1D/ADCY4/PRKCD/RGS2/RGS19/GJB6/GRK6/ARRB2/GNB2/SFN/FKBP1A/GJB2/GNAI2/GRK5/RGS14/PRKACA/SLC8A1/ANXA6/CALM1/GNGT1/YWHAB/CAMK4/RGS4/RGS7/GNG12/CACNA1D/GNB1/GNG2/CACNA1B/RGS3/PRKCB/GNG5/CAMK2B/GNAO1/ATP2B2/CACNA1A/ADCY9/GNG7/GNG2/ATP2A3/RGS7/CAMK2A/PRKCG/GJC1/CHRM2/RGS9/GNAO1/GJB3/YWHAZ/CHRM4/GJD2/KCNJ5/ATP2B3
## WP2380                                                                                                                                                                                                                                 NCF1/MARCKS/SORT1/MARCKS/SIRPA/PRKCD/MAPK3/MAP3K1/APC/MAP2K5/MAPK14/CAMK4/MAPK1/NCAM1/MAP3K1/CFL1/ALPL/MAPK7/RPS6KA1/NCAM1/FYN/PDPK1/MAP3K2/CRTC1/PRKAA2/IGF2BP1/IKBKG/NCAM1/PRKAA1/MAP3K2/RAF1/GRIA1/AKT1/BMP2/KSR1/SH2B2/IGF2BP1/STAT3/CAMK2A/MAPK1/PDPK1/TIAM1/PTK2B/GSK3B/CNR1/SHC3/FOXO3
## WP51                                                                                                                                                                                                                          CD14/PXN/WAS/PIK3CG/MAPK3/PAK1/VCL/RAC2/CYFIP2/ARHGEF1/FGFR4/ARPC5/VCL/FGF14/APC/ACTB/PFN1/IQGAP1/BAIAP2/VIL1/PTK2/MAPK1/CFL1/ITGA1/RASSF7/BAIAP2/SSH2/GNG12/VAV1/SSH3/MOS/FGF17/RAF1/RAC2/SOS1/KRAS/FGF5/PIK3R5/RHOA/MSN/RRAS/PIK3CD/PDGFRA/ARHGEF7/CSK/PAK4/CHRM2/MAPK1/SSH1/BDKRB1/FGF19/FGF8/CHRM4
## WP1449                                                                                                                                                                                                                                             CD14/CCL5/TREM1/CD86/PTPN6/PIK3CG/TLR2/MAPK3/TLR1/MAP2K3/IFNA16/CCL4/TRAFD1/TLR5/MAPK14/MAPK1/CYLD/LY96/IRF7/IRF7/MAPK13/CYLD/IKBKG/TLR6/PLK1/MAP2K6/PELI2/IFNA2/AKT1/PIK3R5/IFNA10/PIK3CD/AKT2/RBCK1/MAPK1/IFNA7/IFNA21/IRAK2/SIGIRR/TOLLIP/IFNA14/TLR7/MYD88/TRAF3/IRAK3/SQSTM1
## WP411               SRPK2/CDC40/TXNL4A/SMC1A/SNRPG/METTL3/NXF1/CPSF4/PTBP1/DHX38/RNGTT/SNRPD2/TMED10/SNRNP70/NONO/FUS/HNRNPM/SNRPA/LSM2/PABPN1/SNRPN/SPOP/SNU13/CPSF2/PRPF4/TMED10/SNRPB2/CSTF1/YBX1/CELF4/SNRPD3/SRSF2/NCBP2/CSTF2/RNPS1/PRPF40A/SF3B1/PPM1G/NCBP1/SRSF3/DNAJC8/SMC1A/NCBP1/SNU13/HNRNPA1/SRSF10/PRPF4B/LSM7/HNRNPC/CLK1/HNRNPAB/SNRPE/PRPF40A/SNRNP40/PTBP2/SRSF6/CSTF3/XRN2/FUS/SNRPE/NCBP1/DDX1/DHX9/HNRNPR/SRSF7/HNRNPD/SF3B3/HNRNPR/NCBP2/RBMX/HNRNPU/SNRPA1/SREK1/SRSF1/SNRPD1/PRMT1/TRA2B/CSTF2T/DHX15/CELF4
## WP179                                                                                                                                                                       ANAPC13/CDC16/SMC1A/CDC23/MCM4/CCND2/CDC14A/ESPL1/ORC4/CUL1/BUB3/ANAPC10/SMAD2/E2F3/ANAPC4/PTTG2/CUL1/CDC6/STAG1/CCNH/PTTG1/TTK/CCNA2/SMC3/MCM2/CHEK2/CCNE1/TFDP2/PRKDC/CDC27/SKP1/SMAD4/CDC7/STAG1/CCNB2/CDK1/YWHAE/ANAPC5/CCNB1/ANAPC7/MCM6/DBF4/ORC5/WEE1/CDC25A/CDC45/ANAPC1/ATR/CDC27/ORC6/BUB1/RB1/CDKN2A/CHEK1/CDK6/MYC/CHEK1/CDK6/CDK6/CDK6/E2F5
## WP3915                                                                                                                                                                                                   RXRA/FBP1/PIK3CG/MAPK3/CBL/MAP3K3/CAP1/MAP3K1/CBL/RPS6KA2/LPL/MAP4K4/MAP2K3/MAP2K5/FLOT2/MAP3K11/SLCO1C1/MAPK14/MAPK1/CBL/MAP3K1/MAPK7/RAPGEF1/RPS6KA1/GYS1/MAPK13/PDPK1/MAP3K2/PRKAA2/PRKAA1/MAP3K2/RAF1/MAP2K6/MAP3K3/SOS1/AKT1/G6PC/IRS4/PIK3CD/AKT2/ANGPTL8/RPS6KA2/MAPK1/PDPK1/GSK3B/SHC3/FOXO3/MAP4K2/MAPK4/SHC1/RHOQ
## WP4217                                                                                                                                                                                                                                                                                                                                       CLTCL1/HLA-F/HLA-C/CLEC10A/HLA-B/ICAM3/MAPK3/PAK1/FLNA/HLA-E/ACTB/IQGAP1/VPS18/IL4/MAPK1/ITGA3/ITGA1/HLA-G/IRF7/IRF7/CREBBP/EP300/NPC2/TFAP2A/HLA-F/AKT1/CTSB/RHOA/ITGA5/PIK3CD/HLA-DOB
## WP2431                                                                                                                                                                                                                                                                                                       ARG1/LILRB2/LGALS3/LTB/LTB/VCAN/FCGR2A/IL1R1/LTB4R/MAPK3/EPHA4/CXCL1/FKBP1A/TNFSF13/IFNG/IL4/MAPK1/MBP/AIF1/TNFSF13B/EPHA4/EPHA4/PLXNA2/CCL2/RHOA/MMP12/PLA2G5/IL2/GFAP/ACAN/NOS1/MAPK1/ROS1/COL2A1/VCAN/MBP/SOX9/FOXO3
## WP710                                                                                                                                                                                                                                                                                                                          BCL6/TCF7/PIK3CG/WNT2B/PTEN/RAC2/FASLG/BIK/MAP3K1/FRAT1/APC/SOD2/CCND3/MAPK1/RBL2/MAP3K1/WNT10B/TCF7L2/RAC2/SOS1/KRAS/AKT1/PIK3R5/RHOA/G6PC/PIK3CD/WNT1/AKT2/WNT10A/AXIN1/MAPK1/PLAU/GSK3B/SOD2/FOXO3
## WP185                                                                                                                                                                                                                                  ITGAM/ITGB2/PXN/VASP/ITGAX/PAK1/ITGAL/VCL/RAC2/CAPN3/CAPN3/VCL/GIT2/MAP2K3/MAP2K5/GIT2/PTK2/MAPK1/ITGA3/MAPK7/RAPGEF1/ITGA1/FYN/PDPK1/TLN1/ITGA8/ZYX/RAF1/MAP2K6/RAC2/SOS1/AKT1/ITGB7/ITGA5/AKT2/ZYX/ILK/ARHGEF7/ARAF/CSK/PAK4/CAV3/CAPNS1/MAPK1/PDPK1/SHC3/MAPK4/CAPN1/ITGA8/SHC1/CAPN9/PAK3
## WP75                                                                                                                                                                                                                                                                                                                       CD14/CCL5/CD86/PIK3CG/TLR2/MAPK3/TLR1/MAP2K3/IFNA16/CCL4/TLR5/MAPK14/MAPK1/LY96/IRF7/IRF7/MAPK13/IKBKG/TLR6/MAP2K6/IFNA2/AKT1/PIK3R5/IFNA10/PIK3CD/AKT2/MAPK1/IFNA7/IFNA21/TOLLIP/IFNA14/TLR7/MYD88/TRAF3
## WP23                                                                                                                                                                                                                                                                                                                                    BCL6/PRKCD/HCLS1/PTPN6/PIK3CG/CBL/RAC2/MAX/CBL/LYN/BLK/PLCG2/PTPN18/ETS1/MAPK14/MAPK1/CBL/PTPRC/SH3BP2/RAPGEF1/LCK/RPS6KA1/VAV1/FYN/PDPK1/IKBKG/DAPP1/PRKCB/RAF1/MAP2K6/RAC2/SOS1/CD79A/AKT1
## WP2806                                                                                                                                                                                                                                                                                                                                                                                           FCN1/FPR1/FCN2/ITGB2/FCGR3A/ELANE/CFP/CFD/CR1/PLAUR/C3AR1/C5AR1/ITGAX/WAS/CR1/TLR2/ARRB2/SELPLG/CR1/C9/GNAI2/CR1/C5AR2/MASP2/PRKACA
## WP231                                                                                                                                                                                                                                                                                                                       TNFRSF1B/PYGL/CYBA/MAPK3/APAF1/MAP3K3/MAP3K1/CASP7/APAF1/PYGL/TNFRSF1A/MAP2K3/NFKBIB/TANK/MAPK1/MAP3K1/TRAF1/CREBBP/TRADD/RIPK3/IKBKG/BCL2L1/PLK1/RAF1/MAP2K6/MAP3K3/CFLAR/SOS1/KRAS/AKT1/CCL2/KSR1/RFFL
## WP4263                                                                                                                                                                                                                                                                                                                                                    TGFA/PRKCD/MAPK3/PAK1/RAC2/RIPK4/TGFBR2/VEGFA/MAPK1/TGFBR1/IKBKG/RALB/BCL2L1/RAF1/JAK1/RAC2/KRAS/AKT1/RHOA/PIK3CD/AKT2/STAT3/ARAF/PAK4/PLD1/TGFBR1/MAPK1/TIAM1/RIPK4/DUSP6
## WP2446                                                                                                                                                                        NPAT/SMARCA2/E2F1/MCM3/RFC3/POLE/MCM7/KIF4A/DNMT1/PCNA/RBBP4/SIN3A/BARD1/TFDP1/ANLN/SIN3A/RBP1/SMC1A/PLK4/MCM4/POLA1/E2F3/SAP30/TTK/CCNA2/SMC3/CCNE1/RRM1/RFC5/TYMS/TFDP2/PRKDC/RFC4/CDC7/CCNB2/BARD1/CDK1/RBBP7/RPA3/CCDC6/DCK/RRM2/CCNB1/STMN1/MCM6/HMGB1/WEE1/CDC25A/MSH6/CDC45/PRIM1/SMC2/RB1/TOP2A/POLE2/CHEK1/CDK6/MYC/CHEK1/CDK6/HLTF/CDK6/CDK6
## WP2328                                                                                                                                                                                                                                                                                                                                                                  LRRK2/LRRK2/HLA-F/GZMB/CD86/HLA-C/HLA-B/FASLG/GNLY/HLA-E/C9/CASP7/GNLY/VEGFA/CD28/CXCR5/IFNG/IL4/HLA-G/PRF1/HLA-F/HARS/CTLA4/COL5A1/HLA-DOB/IL2/PDGFRA/AGTR1
## WP69                                                                                                                                                                                                                                                                                                                                                                                        PRKCD/PSTPIP1/CD3D/WAS/FYB1/MAPK3/PAK1/CBL/CBL/CD28/ITK/LCP2/TNFRSF9/MAPK14/MAPK1/CD3G/CBL/VAV1/LCP2/FYN/PDPK1/IKBKG/RAF1/SOS1/DBNL/AKT1
## WP2059                                                                                                                                                                                                                                                                                 MME/LRP1/MME/MME/CACNA1D/MAPK3/APAF1/CASP7/LPL/APAF1/TNFRSF1A/CALM1/MAPK1/PLCB2/PSEN1/CACNA1D/PSEN1/GAPDH/CALML3/ATP2A3/NCSTN/SNCA/NOS1/MAPK1/GSK3B/CAPN1/GRIN2C/ATF6/FAS/BID/GRIN2A/APBB1/CAPN2/APP/APOE/CACNA1F/PPP3R2/LINC00461/CHP1/CALM3
## WP4298                                                                                                                                                                                                                                                                                                                                                                              ITGB2/MAPK3/NOD2/ITGAL/RAC2/PTCRA/CASP7/CCR3/CASP1/ACTB/TLR5/IFNG/MAPK1/AIF1/FYN/SGCD/BCL2L1/JAK1/RAC2/SOS1/AKT1/KRT8/IL2/STAT3/NOS1/MAPK1/GSK3B
## WP2037                                                                                                                                                                                                                                                                                                                                                   PXN/SIRPA/PTPN6/PIK3CG/MAPK3/PAK1/CBL/FLNA/CTSD/CBL/RPS6KA2/NFKBIB/MAPK14/PTK2/MAPK1/CBL/VAV1/FYN/PPIB/RAF1/JAK1/SOS1/AKT1/AGAP2/ZAP70/STAT3/RPS6KA2/MAPK1/YWHAZ/GSK3B/PPIA
## WP2324                                                                                                                                                                                                                                                                                                                                                                     LGALS3/NCF1/PRKCD/MAPK3/AGER/MAPK14/MAPK1/INHBB/MSR1/ALPL/PRKCB/RAF1/AKT1/RHOA/MSN/STAT3/MAPK1/MYD88/HIF1A/SHC1/SP1/AGER/MAP2K1/RHOA/RELA/RAC1/EGFR/PRKCZ
## WP24                                                                                                                                                                                                                                                                                                                     FPR1/CXCR2/FPR2/CXCR1/CCR1/ATP8A1/C3AR1/C5AR1/OPRL1/CCR3/CCR7/CXCR5/CCR6/AGTR2/NMBR/SSTR3/CCKAR/NPY1R/MC2R/TACR2/CCKBR/ACKR1/HCRTR1/SSTR4/AGTR1/OPRL1/CXCR6/AVPR1B/BDKRB1/AVPR2/GHSR/AGTR2/GALR1/MC1R/SSTR2
## WP3863                                                                                                                                                                                                                                                                                                                                                                                              CD3D/PTPN6/MAPK3/PAK1/CBL/CBL/CD28/ITK/LCP2/CALM1/IFNG/IL4/CBL/PTPRC/LCK/LCP2/FYN/IKBKG/RAF1/SOS1/AKT1/CTLA4/RRAS/IL2/CSF2/ZAP70
## WP45                                                                                                                                                                                                                                                                                                                          MCM3/ORC2/POLE/MCM7/CREB1/PCNA/CREB3/ORC3/TFDP1/MCM4/CCND2/ORC4/PRIM2/E2F3/CCNH/MCM2/CCNE1/TFDP2/MNAT1/CDK1/RPA3/CCNB1/MCM6/ORC5/WEE1/CDC25A/CDC45/PRIM1/ORC6/RB1/POLE2/CDKN2A/CDK6/MYC/CDK6/CDK6/CDK6
## WP304                                                                                                                                                                                                                                                                                                                                                                                                                  PTPN6/MAPK3/CBL/CBL/LYN/MAPK14/MAPK1/CBL/RPS6KA1/VAV1/FYN/EP300/PRKCB/RAF1/SOS1/AKT1/KITLG/SH2B2/INPP5D/STAT3
## WP195                                                                                                                                                                                                                                                                                                             IL1R1/MAPK3/MAP3K3/MAP3K1/MAP2K3/NFKBIB/MAPK14/MAPK1/MAP3K1/MAP3K2/IKBKG/MAP3K2/MAP2K6/MAP3K3/PELI2/HSPB2/AKT1/CCL2/IL1RAP/MAPK1/IRAK2/TOLLIP/MYD88/IRAK3/SQSTM1/MAP2K1/RELA/PELI2/TAB1/PRKCZ/MAP2K6/MAP2K2/IL1RAP
## WP3945                                                                                                                                                                                                                                                                                                                                    ITGAM/TYROBP/IGSF6/ITGB2/SLC7A7/TNFRSF1B/HLX/TCIRG1/TCIRG1/IL10RA/PPP1R18/HCLS1/ITGAX/BIN2/CAPG/KCNE3/RBM47/IL13RA1/FKBP15/KCNE3/ADAP2/APBB1IP/CD37/TGFBR1/RNASE6/RPS6KA1/ELF4/SAMSN1/NPC2
## WP395                                                                                                                                                                                                                                                                                                                                                                                        IL4R/PTPN6/MAPK3/CBL/FLNA/FES/CBL/TYK2/MAPK14/IL4/MAPK1/CBL/EP300/CEBPA/NFIL3/IL2RG/JAK1/SOS1/AKT1/PIK3CD/CEBPB/INPP5D/STAT3/MAPK1/HRH1
## WP585                                                                                                                                                                                                                                                                                                                                                                                                                   PTPN6/CBL/MAP3K1/CBL/TYK2/MAP2K3/MAPK14/CBL/MAP3K1/PTPRC/RAPGEF1/LCK/VAV1/FYN/JAK1/MAP2K6/PIK3CD/ZAP70/STAT3
## WP4205                                                                                                                                                                                                                                                                                                                                                                                                                     MAPK3/PAK1/CBL/CBL/ETS1/PTK2/MAPK1/CBL/RAPGEF1/RAF1/SOS1/KRAS/AKT1/TFE3/PIK3CD/AKT2/STAT3/ARAF/PAK4/MAPK1
## WP2637                                                                                                                                                                                                                                                                                                                                                                                        IL1R1/MAPK3/MAP3K3/MAP3K1/MAP2K3/NFKBIB/TANK/MAPK14/MAPK1/MAP3K1/MBP/IRF7/IRF7/MAP2K6/MAP3K3/MKNK1/IL1RAP/MAPK1/IRAK2/MBP/TOLLIP/MYD88
## WP268                                                                                                                                                                                                                                                                                                                                                                                        DTX4/PTCRA/HES1/PSEN1/CREBBP/PSEN1/JAG1/DTX2/NOTCH1/NOTCH3/APH1B/NCSTN/NUMB/DTX1/INPP5K/KCNJ5/NUMBL/CREBBP/NOTCH2/JAG2/MAML1/RBPJL/DTX3
## WP286                                                                                                                                                                                                                                                                                                                                                                                                 CD86/PTPN6/MAPK3/CBL/CBL/CCR3/LYN/PRKACA/YWHAB/MAPK1/ENPP3/CBL/RAPGEF1/VAV1/FYN/BCL2L1/RAF1/JAK1/SOS1/AKT1/PIK3CD/INPP5D/STAT3
## WP466                                                                                                                                                                                                                                                                                                                                                                        RFC3/ORC2/POLE/MCM7/PCNA/RFC1/ORC3/MCM4/POLA1/ORC4/PRIM2/CDC6/MCM2/RFC5/POLD2/RFC4/CDC7/RFC1/RPA3/GMNN/MCM6/DBF4/ORC5/RFC2/CDC45/PRIM1/ORC6/MCM10/POLE2
## WP3941                                                                                                                                                                                                                                                                                                                                                                                                                                                              TNFRSF1B/C3AR1/C5AR1/APAF1/MAP3K1/APAF1/MAP3K1/TRAF1/MAPK13/TNK2
## WP107                                                                                                                                                                                                                                                                                                                                                    EIF2B5/PAIP1/EIF4EBP1/EIF2B1/EIF3I/EIF3G/EIF3C/EIF2AK3/EEF1B2/EIF2B2/GSPT2/EIF2S3/EEF1A1/EIF5/CLUH/EIF5A/EIF4E/EIF2S1/EIF4A2/EIF4A2/EIF2B3/EIF5B/EIF3H/EIF2AK1/EIF3J/EIF1AY
## WP1591                                                                                                                                                                                                                                                                                                                                                                                                    HAND2/VEGFA/HEY1/MAPK1/FOXH1/PITX2/FOXC1/BHLHE40/BMP10/NOTCH1/BMP2/MAPK1/FGF8/SMYD1/HAND2/TBX5/ERBB3/TBX5/TBX2/IRX4/NFATC4
## WP364                                                                                                                                                                                                                                                                                                                                                                            PRKCD/IL6R/IL6R/MAPK3/TYK2/CRP/MAPK1/VAV1/CREBBP/BCL2L1/JAK1/SOS1/AKT1/STAT3/MAPK1/GSK3B/SHC1/CREBBP/MAP2K1/IRF1/RAC1/MAP2K2/SHC1/TIMP1/HDAC1/SOCS3
## WP2848                                                                                                                                                                                                                                                                                                                                                                                                                                        CXCR1/INHBA/IL6R/IL6R/WNT2B/VEGFA/CSF1R/NT5E/NOTCH1/IGF1/KITLG/CNTF/WNT1/NOG/FGF8/CSF1
## WP2406                                                                                                                                                                                                                                                                                                                                                                                      INHBA/ANPEP/SIRPA/POU5F1/MAPK14/NCAM1/NRG1/NRG1/NRG1/NCAM1/NRG1/NCAM1/THY1/NOTCH1/TBXT/IGF1/PDGFRA/TNNT2/NOG/SOX17/ROR2/SCN5A/IGF2/GSK3B
## WP383                                                                                                                                                                                                                                                                                                                                                                                                                                                   TNNT1/TNNT3/TNNI1/TPM2/DES/MYL9/MYOM1/MYBPC3/MYL4/TPM3/TTN/ACTN2/TNNT2/TNNI2
## WP49                                                                                                                                                                                                                                                                                                                                                                                          IL2RB/MAPK3/CBL/CBL/MAPK1/CBL/LCK/FYN/IL2RG/RAF1/JAK1/SOS1/AKT1/IL2/STAT3/MAPK1/PTK2B/FOXO3/SHC1/MAP2K1/JAK3/MAP2K2/CBL/IL2/SHC1/SOCS3
## WP4136                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                ITGAM/TYROBP/CD14/ITGB2/RASSF5
## WP127                                                                                                                                                                                                                                                                                                                                                                                                   PIK3CG/MAPK3/LYN/MAPK1/RPS6KA1/RAF1/JAK1/SOS1/KRAS/AKT1/IL2/STAT3/MAPK1/GSK3B/FOXO3/SHC1/MAP2K1/CSF2RB/MAP2K2/IL2/SHC1/IL5RA
## WP3937                                                                                                                                                                                                                                                                                                                                                                                                                     ITGAM/TYROBP/CYBB/NCF1/ITGB2/TREM1/FCER1G/CYBA/ARPC1B/PTPN6/NCF4/NCF4/PIK3CG/RAC2/FCGR1A/FCGR1A/LYN/PLCG2
## WP2525                                                                                                                                                                                                                                                                                                                                                                                                      AHCYL2/MTR/SHMT2/AHCY/PSPH/TYMS/MTHFD2L/CTH/MTHFD1/PSAT1/DNMT3A/MTHFD2L/BHMT/PHGDH/MTHFD2/CBS/GCLC/AHCYL1/MTHFD1L/DNMT3B
## WP313                                                                                                                                                                                                                                                                                                                                                                                                                                                       PXN/MAPK3/PAK1/PTEN/PTK2/MAPK1/RAPGEF1/ITGA1/RAF1/SOS1/STAT3/MAPK1/PTK2B
## WP453                                                                                                                                                                                                                                                                                                                                                                                                                                           TNFRSF1B/IL2RB/IL4R/CD86/TNFRSF1A/CD28/IFNG/IL4/LCK/IL2RG/LAMB2/COL3A1/IL2/ZAP70/VTN
## WP422                                                                                                                                                                                                                                                                                                                                                                                                             MAPK3/MAP3K3/SIPA1/MAP3K1/MAP2K3/MAPK14/MAPK1/MAP3K1/MBP/MAP3K2/MAP3K2/RAF1/MAP2K6/MAP3K3/KRAS/RRAS/ARAF/MAPK1/MBP
## WP581                                                                                                                                                                                                                                                                                                                                                                                                 PIK3CG/MAPK3/MAPK1/PTPRC/PTPRU/RAF1/SOS1/AKT1/STAT3/MAPK1/SHC1/MAP2K1/MAP2K2/SHC1/SRC/PDK1/EPOR/AKT1/RASA1/SOCS1/EPO/IRS2/JAK2
## WP4197                                                                                                                                                                                                                                                                                                                                                                                                                                     IFITM1/IFI35/TYK2/OAS1/IFNGR1/IFIT3/JAK1/IRF9/IRF1/IFIT1/TAP1/SOCS1/IFIT3/OAS1/PSMB8/JAK2
## WP4301                                                                                                                                                                                                                                                                                                                                                                                                                                                                        MAPK3/RAB27A/MAPK1/RAB5C/RAF1/KRAS/RRAS/ARAF/HGS/MAPK1
## WP3678                                                                                                                                                                                                                                                                                                                                                                                                                                                                          TCF7/PIK3CG/VEGFA/POSTN/JAG1/NOTCH1/TCF7L2/EPAS1/TNC</code></pre>
<pre class="r"><code>gset@featureData@data[which(gset@featureData@data[,&quot;Gene.ID&quot;]==100128545),c(&quot;ID&quot;,&quot;Gene.ID&quot;)]</code></pre>
<pre><code>##                      ID   Gene.ID
## 10012683009 10012683009 100128545</code></pre>
</div>
<div id="the-following-is-the-default-plots-as-pulled-by-geo2r-scripts" class="section level4">
<h4>The following is the default plots, as pulled by Geo2R scripts…</h4>
<pre class="r"><code>################################################################
#   Boxplot for selected GEO samples
#library(Biobase)
#library(GEOquery)

# load series and platform data from GEO

suppressMessages(gset &lt;- getGEO(&quot;GSE4170&quot;, GSEMatrix =TRUE, getGPL=FALSE))
if (length(gset) &gt; 1) idx &lt;- grep(&quot;GPL2029&quot;, attr(gset, &quot;names&quot;)) else idx &lt;- 1
gset &lt;- gset[[idx]]

# group names for all samples in a series
gsms &lt;- paste0(&quot;000000111111111111222222222222222X22222X2222122122&quot;,
        &quot;22555555555555555555555552222255555555555555555555&quot;,
        &quot;5555555555555522222&quot;)
sml &lt;- c()
for (i in 1:nchar(gsms)) { sml[i] &lt;- substr(gsms,i,i) }
sml &lt;- paste(&quot;G&quot;, sml, sep=&quot;&quot;) #set group names

# eliminate samples marked as &quot;X&quot;
sel &lt;- which(sml != &quot;X&quot;)
sml &lt;- sml[sel]
gset &lt;- gset[ ,sel]

# order samples by group
ex &lt;- exprs(gset)[ , order(sml)]
sml &lt;- sml[order(sml)]
fl &lt;- as.factor(sml)
labels &lt;- c(&quot;cd&quot;,&quot;ap&quot;,&quot;bc&quot;,&quot;resistant&quot;,&quot;responder&quot;,&quot;chronic&quot;)

# set parameters and draw the plot
palette(c(&quot;#dfeaf4&quot;,&quot;#f4dfdf&quot;,&quot;#f2cb98&quot;,&quot;#dfeaf4&quot;,&quot;#f4dfdf&quot;,&quot;#dfeaf4&quot;, &quot;#AABBCC&quot;))
dev.new(width=4+dim(gset)[[2]]/5, height=6)
par(mar=c(2+round(max(nchar(sampleNames(gset)))/2),4,2,1))
title &lt;- paste (&quot;GSE4170&quot;, &#39;/&#39;, annotation(gset), &quot; selected samples&quot;, sep =&#39;&#39;)
boxplot(ex[,1:20], boxwex=0.6, notch=T, main=title, outline=FALSE, las=2, col=fl) #ben: only take first 20 columns
legend(&quot;topleft&quot;, labels, fill=palette(), bty=&quot;n&quot;)</code></pre>
<pre class="r"><code># set parameters and draw the plot
palette(c(&quot;#dfeaf4&quot;,&quot;#f4dfdf&quot;,&quot;#f2cb98&quot;, &quot;#AABBCC&quot;))
dev.new(width=4+dim(gset)[[2]]/5, height=6)
par(mar=c(2+round(max(nchar(sampleNames(gset)))/2),4,2,1))
title &lt;- paste (&quot;GSE4170&quot;, &#39;/&#39;, annotation(gset), &quot; selected samples&quot;, sep =&#39;&#39;)
boxplot(ex[,0:20], boxwex=0.6, notch=T, main=title, outline=FALSE, las=2, col=fl) #ben: only take first 20 columns
legend(&quot;topleft&quot;, labels, fill=palette(), bty=&quot;n&quot;)</code></pre>
</div>
</div>
<div id="step-2-repeat-the-analysis-for-the-other-study" class="section level3">
<h3>Step 2: Repeat the analysis for the other study</h3>
<pre class="r"><code># Version info: R 3.2.3, Biobase 2.30.0, GEOquery 2.40.0, limma 3.26.8
# R scripts generated  Thu Mar 28 05:38:05 EDT 2019

################################################################
#   Differential expression analysis with limma
library(Biobase)
library(GEOquery)
library(limma)

# load series and platform data from GEO

gset &lt;- getGEO(&quot;GSE14671&quot;, GSEMatrix =TRUE, AnnotGPL=TRUE)</code></pre>
<pre><code>## Found 1 file(s)</code></pre>
<pre><code>## GSE14671_series_matrix.txt.gz</code></pre>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_double(),
##   ID_REF = col_character()
## )</code></pre>
<pre><code>## See spec(...) for full column specifications.</code></pre>
<pre><code>## File stored at:</code></pre>
<pre><code>## /var/folders/_6/z0dk_wj97396pxknwtg8gmbc0000gn/T//RtmpA3UT3o/GPL570.annot.gz</code></pre>
<pre><code>## Warning: 62 parsing failures.
##   row             col           expected    actual         file
## 54614 Platform_SPOTID 1/0/T/F/TRUE/FALSE --Control literal data
## 54615 Platform_SPOTID 1/0/T/F/TRUE/FALSE --Control literal data
## 54616 Platform_SPOTID 1/0/T/F/TRUE/FALSE --Control literal data
## 54617 Platform_SPOTID 1/0/T/F/TRUE/FALSE --Control literal data
## 54618 Platform_SPOTID 1/0/T/F/TRUE/FALSE --Control literal data
## ..... ............... .................. ......... ............
## See problems(...) for more details.</code></pre>
<pre class="r"><code>if (length(gset) &gt; 1) idx &lt;- grep(&quot;GPL570&quot;, attr(gset, &quot;names&quot;)) else idx &lt;- 1
gset &lt;- gset[[idx]]

# make proper column names to match toptable 
fvarLabels(gset) &lt;- make.names(fvarLabels(gset))

# group names for all samples
gsms &lt;- &quot;33333333333344444444444444444444444433333344444444444444444&quot;
sml &lt;- c()
for (i in 1:nchar(gsms)) { sml[i] &lt;- substr(gsms,i,i) }

# log2 transform
ex &lt;- exprs(gset)
qx &lt;- as.numeric(quantile(ex, c(0., 0.25, 0.5, 0.75, 0.99, 1.0), na.rm=T))
LogC &lt;- (qx[5] &gt; 100) ||
          (qx[6]-qx[1] &gt; 50 &amp;&amp; qx[2] &gt; 0) ||
          (qx[2] &gt; 0 &amp;&amp; qx[2] &lt; 1 &amp;&amp; qx[4] &gt; 1 &amp;&amp; qx[4] &lt; 2)
if (LogC) { ex[which(ex &lt;= 0)] &lt;- NaN
  exprs(gset) &lt;- log2(ex) }

# set up the data and proceed with analysis
sml &lt;- paste(&quot;G&quot;, sml, sep=&quot;&quot;)    # set group names
fl &lt;- as.factor(sml)
gset$description &lt;- fl
design &lt;- model.matrix(~ description + 0, gset)
colnames(design) &lt;- levels(fl)
fit &lt;- lmFit(gset, design)
cont.matrix &lt;- makeContrasts(G4-G3, levels=design)
fit2 &lt;- contrasts.fit(fit, cont.matrix)
fit2 &lt;- eBayes(fit2, 0.01)
tT &lt;- topTable(fit2, adjust=&quot;fdr&quot;, sort.by=&quot;B&quot;, number=250)</code></pre>
<pre class="r"><code>tT &lt;- subset(tT, select=c(&quot;ID&quot;,&quot;adj.P.Val&quot;,&quot;P.Value&quot;,&quot;Gene.symbol&quot;,&quot;Gene.title&quot;,&quot;Gene.ID&quot;,&quot;GO.Function&quot;,&quot;GO.Process&quot;))
write.table(head(tT, file=stdout(), row.names=F, sep=&quot;\t&quot;))</code></pre>
<pre><code>## &quot;ID&quot; &quot;adj.P.Val&quot; &quot;P.Value&quot; &quot;Gene.symbol&quot; &quot;Gene.title&quot; &quot;Gene.ID&quot; &quot;GO.Function&quot; &quot;GO.Process&quot;
## &quot;40612_at&quot; &quot;40612_at&quot; 0.886489859828241 4.28338670721571e-05 &quot;DOPEY1&quot; &quot;dopey family member 1&quot; &quot;23033&quot; &quot;&quot; &quot;protein transport&quot;
## &quot;216944_s_at&quot; &quot;216944_s_at&quot; 0.886489859828241 9.75579674293235e-05 &quot;ITPR1&quot; &quot;inositol 1,4,5-trisphosphate receptor type 1&quot; &quot;3708&quot; &quot;calcium channel inhibitor activity///calcium ion transmembrane transporter activity///calcium-release channel activity///inositol 1,4,5-trisphosphate-sensitive calcium-release channel activity///inositol 1,4,5-trisphosphate-sensitive calcium-release channel activity///phosphatidylinositol binding///protein binding&quot; &quot;calcium ion transport///endoplasmic reticulum calcium ion homeostasis///epithelial fluid transport///inositol phosphate-mediated signaling///intrinsic apoptotic signaling pathway in response to endoplasmic reticulum stress///negative regulation of calcium-mediated signaling///platelet activation///post-embryonic development///regulation of autophagy///regulation of cardiac conduction///regulation of insulin secretion///release of sequestered calcium ion into cytosol///response to hypoxia///signal transduction///voluntary musculoskeletal movement&quot;
## &quot;223325_at&quot; &quot;223325_at&quot; 0.886489859828241 0.000132743036337203 &quot;TXNDC11&quot; &quot;thioredoxin domain containing 11&quot; &quot;51061&quot; &quot;protein binding///protein disulfide isomerase activity&quot; &quot;cell redox homeostasis///protein folding///response to endoplasmic reticulum stress&quot;
## &quot;228411_at&quot; &quot;228411_at&quot; 0.886489859828241 0.00013866443214503 &quot;PARD3B&quot; &quot;par-3 family cell polarity regulator beta&quot; &quot;117583&quot; &quot;&quot; &quot;cell cycle///cell division&quot;
## &quot;1556346_at&quot; &quot;1556346_at&quot; 0.886489859828241 0.000150695724736774 &quot;COTL1&quot; &quot;coactosin like F-actin binding protein 1&quot; &quot;23406&quot; &quot;actin binding///enzyme binding///protein binding&quot; &quot;biological_process///defense response to fungus&quot;
## &quot;227314_at&quot; &quot;227314_at&quot; 0.886489859828241 0.000154158404755628 &quot;ITGA2&quot; &quot;integrin subunit alpha 2&quot; &quot;3673&quot; &quot;collagen binding///collagen binding involved in cell-matrix adhesion///collagen receptor activity///integrin binding///laminin binding///metal ion binding///protein binding///protein complex binding///protein heterodimerization activity///virus receptor activity&quot; &quot;blood coagulation///cell adhesion///cell adhesion mediated by integrin///cell proliferation///cell-matrix adhesion///cell-substrate adhesion///cellular response to estradiol stimulus///cellular response to mechanical stimulus///collagen-activated signaling pathway///detection of mechanical stimulus involved in sensory perception of pain///establishment of protein localization///extracellular matrix organization///female pregnancy///focal adhesion assembly///hepatocyte differentiation///hypotonic response///integrin-mediated signaling pathway///mammary gland development///mesodermal cell differentiation///organ morphogenesis///positive regulation of DNA binding///positive regulation of alkaline phosphatase activity///positive regulation of cell adhesion///positive regulation of cell projection organization///positive regulation of collagen binding///positive regulation of collagen biosynthetic process///positive regulation of epithelial cell migration///positive regulation of inflammatory response///positive regulation of leukocyte migration///positive regulation of phagocytosis, engulfment///positive regulation of positive chemotaxis///positive regulation of smooth muscle cell migration///positive regulation of smooth muscle cell proliferation///positive regulation of smooth muscle contraction///positive regulation of translation///positive regulation of transmission of nerve impulse///response to L-ascorbic acid///response to amine///response to drug///response to hypoxia///response to muscle activity///response to parathyroid hormone///skin morphogenesis///substrate-dependent cell migration///viral entry into host cell&quot;</code></pre>
<p>the “profile graph” didn’t quite work as nicely (the Radich study, just had to enter one of the internal gene IDs). Here, it required opening a spearate file to look for the ID to profile.<br />
</p>
<p>“This tab allows you to view a specific gene expression profile graph by entering the corresponding identifier from the ID column of the Platform record. This feature does not perform any calculations; it merely displays the expression values of the gene across Samples. Sample groups may or may not be defined for this feature to work.”</p>
<pre class="r"><code>################################################################
#   Boxplot for selected GEO samples
library(Biobase)
library(GEOquery)

# load series and platform data from GEO

gset &lt;- getGEO(&quot;GSE14671&quot;, GSEMatrix =TRUE, getGPL=FALSE)</code></pre>
<pre><code>## Found 1 file(s)</code></pre>
<pre><code>## GSE14671_series_matrix.txt.gz</code></pre>
<pre><code>## Using locally cached version: /var/folders/_6/z0dk_wj97396pxknwtg8gmbc0000gn/T//RtmpA3UT3o/GSE14671_series_matrix.txt.gz</code></pre>
<pre><code>## Parsed with column specification:
## cols(
##   .default = col_double(),
##   ID_REF = col_character()
## )</code></pre>
<pre><code>## See spec(...) for full column specifications.</code></pre>
<pre class="r"><code>if (length(gset) &gt; 1) idx &lt;- grep(&quot;GPL570&quot;, attr(gset, &quot;names&quot;)) else idx &lt;- 1
gset &lt;- gset[[idx]]

# group names for all samples in a series
gsms &lt;- &quot;33333333333344444444444444444444444433333344444444444444444&quot;
sml &lt;- c()
for (i in 1:nchar(gsms)) { sml[i] &lt;- substr(gsms,i,i) }
sml &lt;- paste(&quot;G&quot;, sml, sep=&quot;&quot;)  #set group names

# order samples by group
ex &lt;- exprs(gset)[ , order(sml)]
sml &lt;- sml[order(sml)]
fl &lt;- as.factor(sml)
labels &lt;- c(&quot;cd&quot;,&quot;ap&quot;,&quot;bc&quot;,&quot;resistant&quot;,&quot;responder&quot;)</code></pre>
<pre class="r"><code># set parameters and draw the plot
palette(c(&quot;#dfeaf4&quot;,&quot;#f4dfdf&quot;,&quot;#f2cb98&quot;,&quot;#dfeaf4&quot;,&quot;#f4dfdf&quot;, &quot;#AABBCC&quot;))
dev.new(width=4+dim(gset)[[2]]/5, height=6)
par(mar=c(2+round(max(nchar(sampleNames(gset)))/2),4,2,1))
title &lt;- paste (&quot;GSE14671&quot;, &#39;/&#39;, annotation(gset), &quot; selected samples&quot;, sep =&#39;&#39;)
boxplot(ex, boxwex=0.6, notch=T, main=title, outline=FALSE, las=2, col=fl)
legend(&quot;topleft&quot;, labels, fill=palette(), bty=&quot;n&quot;)</code></pre>
<p>So, both datasets are loaded. They total less than 80MB each. Would be more interesting to eventually get gene-level info (splits by funcitonal pathway,etc).</p>
<p>But first step is to aggregate the 2 datasets together. Questions to ask: are the ranges and distributions the same for both datasets, roughly? Do the 2 populations look similar in terms of value range or even distribution (bimodal, normal, etc)? *If not very similar… is there similarity WITHIN the populations of one study? or is the population within a study as ~heterogeneious as when compared ‘across’ studies?</p>
<p>+do i need to normalize the ranges or what? to get similaris distributions…?</p>
<p>NExt, are the “top 250” genes identified by the GEO2R default comparison… how similar is the “top 250” lists for the 2 studies? *This will lead naturally into a categorization of the genes in the 3 main pathways of interest. May also want to see highly expressed genes vs. normal patient levels (which should… highlight more genes that are members of the 3 pathways, id think)</p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
